% Template for PLoS
% Version 3.6 Aug 2022
%
% % % % % % % % % % % % % % % % % % % % % %
%
% -- IMPORTANT NOTE
%
% This template contains comments intended
% to minimize problems and delays during our production
% process. Please follow the template instructions
% whenever possible.
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% Once your paper is accepted for publication,
% PLEASE REMOVE ALL TRACKED CHANGES in this file
% and leave only the final text of your manuscript.
% PLOS recommends the use of latexdiff to track changes during review, as this will help to maintain a clean tex file.
% Visit https://www.ctan.org/pkg/latexdiff?lang=en for info or contact us at latex@plos.org.
%
%
% There are no restrictions on package use within the LaTeX files except that no packages listed in the template may be deleted.
%
% Please do not include colors or graphics in the text.
%
% The manuscript LaTeX source should be contained within a single file (do not use \input, \externaldocument, or similar commands).
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% -- FIGURES AND TABLES
%
% Please include tables/figure captions directly after the paragraph where they are first cited in the text.
%
% DO NOT INCLUDE GRAPHICS IN YOUR MANUSCRIPT
% - Figures should be uploaded separately from your manuscript file.
% - Figures generated using LaTeX should be extracted and removed from the PDF before submission.
% - Figures containing multiple panels/subfigures must be combined into one image file before submission.
% For figure citations, please use "Fig" instead of "Figure".
% See http://journals.plos.org/plosone/s/figures for PLOS figure guidelines.
%
% Tables should be cell-based and may not contain:
% - spacing/line breaks within cells to alter layout or alignment
% - do not nest tabular environments (no tabular environments within tabular environments)
% - no graphics or colored text (cell background color/shading OK)
% See http://journals.plos.org/plosone/s/tables for table guidelines.
%
% For tables that exceed the width of the text column, use the adjustwidth environment as illustrated in the example table in text below.
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% -- EQUATIONS, MATH SYMBOLS, SUBSCRIPTS, AND SUPERSCRIPTS
%
% IMPORTANT
% Below are a few tips to help format your equations and other special characters according to our specifications. For more tips to help reduce the possibility of formatting errors during conversion, please see our LaTeX guidelines at http://journals.plos.org/plosone/s/latex
%
% For inline equations, please be sure to include all portions of an equation in the math environment.  For example, x$^2$ is incorrect; this should be formatted as $x^2$ (or $\mathrm{x}^2$ if the romanized font is desired).
%
% Do not include text that is not math in the math environment. For example, CO2 should be written as CO\textsubscript{2} instead of CO$_2$.
%
% Please add line breaks to long display equations when possible in order to fit size of the column.
%
% For inline equations, please do not include punctuation (commas, etc) within the math environment unless this is part of the equation.
%
% When adding superscript or subscripts outside of brackets/braces, please group using {}.  For example, change "[U(D,E,\gamma)]^2" to "{[U(D,E,\gamma)]}^2".
%
% Do not use \cal for caligraphic font.  Instead, use \mathcal{}
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% Please contact latex@plos.org with any questions.
%
% % % % % % % % % % % % % % % % % % % % % % % %

\documentclass[10pt,letterpaper]{article}
\usepackage[top=0.85in,left=2.75in,footskip=0.75in]{geometry}

% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

% ligatures disabled
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% color can be used to apply background shading to table cells only
\usepackage[table]{xcolor}

% array package and thick rules for tables
\usepackage{array}

% create "+" rule type for thick vertical lines
\newcolumntype{+}{!{\vrule width 2pt}}

% create \thickcline for thick horizontal lines of variable length
\newlength\savedwidth
\newcommand\thickcline[1]{%
\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
\cline{#1}%
\noalign{\vskip\arrayrulewidth}%
\noalign{\global\arrayrulewidth\savedwidth}%
}

% \thickhline command for thick horizontal lines that span the table
\newcommand\thickhline{\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
\hline
\noalign{\global\arrayrulewidth\savedwidth}}


% Remove comment for double spacing
%\usepackage{setspace}
%\doublespacing

% Text layout
\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}
\renewcommand{\figurename}{Fig}

% Use the PLoS provided BiBTeX style
\bibliographystyle{plos2015}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother



% Header and Footer with logo
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
%\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
%\setlength{\headheight}{27.023pt}
%\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\headrulewidth}{0pt}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}
\lfoot{\today}

%% Include all macros below
\usepackage{mathtools}
\definecolor{LIGHTGREY}{gray}{0.9}
\usepackage{bm}
\usepackage{booktabs}
\usepackage{makecell}

\newcommand{\der}{\text{d}}
\newcommand{\e}{\text{e}}
\newcommand{\Ne}{N_{\text{e}}}
\newcommand{\proba}{\mathbb{P}}
\newcommand{\pfix}{\proba_{\text{fix}}}
\newcommand{\dn}{d_N}
\newcommand{\ds}{d_S}
\newcommand{\dnds}{\dn / \ds}
\newcommand{\Sphy}{S_{0}}
\newcommand{\SphyDel}{\mathcal{D}_0}
\newcommand{\SphyNeu}{\mathcal{N}_0}
\newcommand{\SphyBen}{\mathcal{B}_0}
\newcommand{\Sphyclass}{x}
\newcommand{\SphyclassAlt}{y}
\newcommand{\given}{\mid}
\newcommand{\Spop}{S}
\newcommand{\SpopDel}{\mathcal{D}}
\newcommand{\SpopNeu}{\mathcal{N}}
\newcommand{\SpopBen}{\mathcal{B}}
\newcommand{\ProbaPopDel}{\proba [ \SpopDel]}
\newcommand{\ProbaPopNeu}{\proba [ \SpopNeu ]}
\newcommand{\ProbaPopBen}{\proba [ \SpopBen ]}
\newcommand{\AdvMean}{\beta_b}
\newcommand{\DelMean}{\beta_d}
\newcommand{\thetaSyn}{\theta_{\text{S}}}
%\renewcommand{\baselinestretch}{1.5}
%\renewcommand{\arraystretch}{0.6}

%% END MACROS SECTION

\begin{document}
\vspace*{0.2in}

% Title must be 250 characters or less.
\begin{flushleft}
{\Large
\textbf\newline{Estimating the proportion of beneficial mutations that are not adaptive in mammals} % Please use "sentence case" for title and headings (capitalize only the first word in a title (or heading), the first word in a subtitle (or subheading), and any proper nouns).
}
\newline
% Insert author names, affiliations and corresponding author email (do not include titles, positions, or degrees).
\\
T. {Latrille}\textsuperscript{1\ddag*},
J. {Joseph}\textsuperscript{2\ddag},
D.~A. {Hartasánchez}\textsuperscript{1},
N. {Salamin}\textsuperscript{1}
\\
\bigskip
\textbf{1} Department of Computational Biology, Université de Lausanne, Lausanne, Switzerland
\\
\textbf{2} Laboratoire de Biométrie et Biologie Evolutive, UMR5558, Université Lyon 1, Villeurbanne, France
\\

\bigskip

% Insert additional author notes using the symbols described below. Insert symbol callouts after author names as necessary.
%
% Remove or comment out the author notes below if they aren't used.
%
% Primary Equal Contribution Note
% \Yinyang These authors contributed equally to this work.

% Additional Equal Contribution Note
% Also use this double-dagger symbol for special authorship notes, such as senior authorship.
\ddag These authors also contributed equally to this work.

% Current address notes
% \textcurrency Current Address: Department of Computational Biology, Université de Lausanne, Lausanne, Switzerland % change symbol to "\textcurrency a" if more than one current address note

% Group/Consortium Author Note

% Use the asterisk to denote corresponding authorship and provide email address in note below.
* \href{mailto:thibault.latrille@ens-lyon.org}{thibault.latrille@ens-lyon.org}

\end{flushleft}

\section*{Abstract}
Mutations can be beneficial by bringing innovation to their bearer, allowing them to adapt to environmental change.
These mutations are typically unpredictable since they respond to an unforeseen change in the environment.
However, mutations can also be beneficial because they are simply restoring a state of higher fitness that was lost due to genetic drift in a stable environment.
In contrast to adaptive mutations, these beneficial non-adaptive mutations can be predicted if the underlying fitness landscape is stable and known.
The contribution of such non-adaptive mutations to molecular evolution has been widely neglected mainly because their detection is very challenging.
We have here reconstructed protein-coding-gene fitness landscapes shared between mammals, using mutation-selection models and a multi-species alignments across 87 mammals.
These fitness landscapes have allowed us to predict the fitness effect of polymorphisms found in 28 mammalian populations.
Using methods that quantify selection at the population level, we have confirmed that beneficial non-adaptive mutations are indeed positively selected in extant populations.
Our work confirms that deleterious substitutions are accumulating in mammals and are being reverted, generating a balance in which genomes are damaged and restored simultaneously at different loci.
We observe that beneficial non-adaptive mutations represent between 15\% and 45\% of all beneficial mutations in 24 of 28 populations analyzed, suggesting that a substantial part of ongoing positive selection is not driven solely by adaptation to environmental change in mammals.


\section*{Author summary}
The extent to which adaptation to changing environments is shaping genomes is a central question in molecular evolution.
To quantify the rate of adaptation, population geneticists have typically used signatures of positive selection.
However, mutations restoring an ancestral state of higher fitness lost by genetic drift are also positively selected, but they do not respond to a change in the environment.
In this study, we have managed to distinguish beneficial mutations that are due to changing environments and those that are restoring pre-existing functions in mammals.
We show that a substantial proportion of beneficial mutations cannot be interpreted as adaptive.

\linenumbers


\section*{Introduction}
Adaptation is one of the main processes shaping the diversity of forms and functions across the tree of life~\cite{darwin_origin_1859}.
Evolutionary adaptation is tightly linked to environmental change and species responding to this change~\cite{merrell_adaptive_1994, gavrilets_adaptive_2009}.
Such environmental changes are either abiotic (e.g. temperature, humidity) or biotic (e.g. pressure from predators or viruses~\cite{enard_viruses_2016}).
For adaptation to occur, there must be variation within populations, which mostly appears via mutations in the DNA sequence.
While neutral mutations will not impact an individual's fitness, deleterious mutations have a negative effect, and beneficial mutations improve their bearer's fitness.
A beneficial mutation is thus more likely than a neutral mutation to invade the population and reach fixation, resulting in a substitution at the species level.

Upon environmental change, because adaptive beneficial mutations toward new fitness optima are more likely, the number of substitutions also increases (Fig~\ref{fig:fitness-landscape}A).
An increased substitution rate is thus commonly interpreted as a sign of adaptation~\cite{mcdonald_adaptative_1991, smith_adaptive_2002, welch_estimating_2006}.
The availability of large-scale genomic data and the development of theoretical models have enabled the detection and quantification of substitution rate changes across genes and lineages~\cite{yang_statistical_2000, eyre-walker_genomic_2006, moutinho_variation_2019}.
These approaches, now common practice in evolutionary biology, have helped better understand the processes underpinning the rates of molecular evolution, contributing to disentangling the effects of mutation, selection and drift in evolution~\cite{lynch_mutation_2023}.
However, a collateral effect has been conflating beneficial mutations with adaptive evolution when adaptive evolution is not the only process that can lead to beneficial mutations~\cite{charlesworth_other_2007, mustonen_fitness_2009, jones_shifting_2017}.

\begin{figure}[!htb]
\centering
\includegraphics[width=\textwidth, page=1] {figure1.eps}
\caption{
{\bf Changing and stable fitness landscapes.}
(A \& B) For a given codon position of a protein-coding DNA sequence, amino acids (x-axis) have different fitness values (y-axis).
Under a changing fitness landscape (A), these fitnesses fluctuate with time.
The protein sequence follows the moving target defined by the amino-acid fitnesses. Since substitutions are preferentially accepted if they are in the direction of this target, substitutions are, on average, adaptive.
At the phylogenetic scale (C), beneficial substitutions are common (positive signs), promoting phenotype diversification across species.
Under a stable fitness landscape (B), most mutations reaching fixation are either slightly deleterious reaching fixation due to drift or are beneficial non-adaptive mutations restoring a more optimal amino acid.
At the phylogenetic scale (D), deleterious substitutions (negative signs) are often reverted via beneficial non-adaptive mutations (positive signs), promoting phenotype stability and preserving well-established biological systems.
Even though, individually, any beneficial non-adaptive mutation might have a weak effect on its bearer, we expect them to be scattered across the genome and the genome-wide signature of beneficial non-adaptive mutations to be detectable and quantifiable.}
\label{fig:fitness-landscape}
\end{figure}

\subsection*{Beneficial yet non-adaptive mutations}

In a constant environment, a deleterious mutation can reach fixation by genetic drift~\cite{ohta_nearly_1992}.
A new mutation restoring the ancestral fitness will thus be beneficial (Fig~\ref{fig:fitness-landscape}B), even though the environment has not changed~\cite{gillespie_ohta_1995, hartl_compensatory_1996, sella_application_2005, mustonen_fitness_2009, cvijovic_fate_2015}.
We will refer to as beneficial non-adaptive mutations to those mutations that restore the ancestral fitness under the assumption that the fitness landscape has not changed~\cite{piganeau_estimating_2003, charlesworth_other_2007}.
Such mutations can happen through a mutation at a different locus, in which case it is called a compensatory mutation~\cite{hartl_compensatory_1996, mustonen_fitness_2009}.
While compensatory mutations change the sequence and thus induce genetic diversification, beneficial non-adaptive mutations at the locus of the initial mutation reduce genetic diversity and do not contribute to genetic innovation, which are the focus of this manuscript.
Although Tomoko Ohta considered beneficial non-adaptive mutations negligible in her nearly-neutral theory~\cite{ohta_nearly_1992}, their importance has now been acknowledged for expanding populations~\cite{charlesworth_other_2007}.
However, differentiating between an adaptive mutation and a beneficial non-adaptive mutation remains challenging~\cite{chi_detecting_2020}.
Indeed, an adaptive mutation responding to a change in the environment and a beneficial non-adaptive mutation have equivalent fitness consequences for their bearer~\cite{charlesworth_other_2007}.
Similarly, at the population level, both types of mutations will result in a positive transmission bias of the beneficial allele.
However, at the macro-evolutionary scale, the consequences of these two types of mutations are fundamentally different.
While adaptive mutations promote phenotype diversification (Fig~\ref{fig:fitness-landscape}C), beneficial non-adaptive mutations promote phenotype stability and may help preserve well-established biological systems (Fig~\ref{fig:fitness-landscape}D).
Additionally, the direction of adaptive evolution is unpredictable because it is caused by an unforeseen change in the environment and, hence, in the underlying fitness landscape~\cite{bazykin_changing_2015}.
On the other hand, beneficial non-adaptive mutations are predictable because, under a stable fitness landscape, any change from non-optimal to optimal amino acids will move back the site toward the equilibrium expected under the fitness landscape~\cite{moses_inferring_2009, fischer_germline_2011, chen_hunting_2021}.
They can then be distinguished from truly novel beneficial mutations because the latter are not expected to mutate toward the amino acids of higher fitnesses defined by the stable fitness landscape but rather mutate to amino acids showing a diversified pattern (Fig~\ref{fig:fitness-landscape}).

\subsection*{Fitness landscape reconstruction}

The mutation-selection framework permits to link the patterns of substitution along a phylogenetic tree with the underlying fitness landscape~\cite{halpern_evolutionary_1998, mccandlish_modeling_2014}.
Such mutation-selection models applied to protein-coding DNA sequence alignments at the codon level allow us to estimate relative fitnesses for all amino acids for each site of the sequence, explicitly assuming that the underlying fitness landscape is stable along the phylogenetic tree~\cite{rodrigue_mechanistic_2010, tamuri_estimating_2012, rodrigue_detecting_2017}.
Moreover, effective population size ($\Ne$) is considered constant along the phylogenetic tree precisely because of the fixed fitness landscape assumption, the consequences of which are detailed in the Discussion.
Importantly, because mutation-selection codon models at the phylogenetic scale are based on population-genetics equations, their estimates of selection coefficients are directly interpretable as fitness effects at the population scale; and because they work at the DNA level, we are able to account for mutational bias in DNA and structure of the genetic code.
The model further integrates the shared evolutionary history between samples and their divergence, which, together, allow us to estimate fitness effects in sequence alignments even though sequences are not independent samples and might not represent the equilibrium distribution of amino acids (see section~\ref{subsec:s-phylogeny-method} in Materials \& Methods).
Detailed model description is available in~\nameref{S1_File} (\nameref{S1_Fig}).

Accordingly, fitting the mutation-selection model to a multi-species sequence alignment allows us to obtain relative fitnesses for all amino acids (Fig~\ref{fig:method}A).
The difference in fitness between a pair of amino acids allows us to predict whether any mutation would be a deleterious mutation toward a less fit amino acid, a nearly-neutral mutation, or a mutation toward a known fitter amino acid constituting thus a beneficial non-adaptive mutation (Fig~\ref{fig:method}B).
We can hence use large-scale genomic data to test whether such fitnesses estimated at the phylogenetic scale predict the fitness effects at the population scale.
The placental mammals represent an excellent study system to perform such an analysis.
Having originated $\sim$102 million years ago, they diversified quickly~\cite{foley_genomic_2023}.
Additionally, polymorphism data are available for many species~\cite{howe_ensembl_2021}, as are high quality protein-coding DNA alignments across the genome~\cite{ranwez_orthomam_2007, scornavacca_orthomam_2019}.
By performing our analysis on 14,509 orthologous protein-coding genes across 87 species, we focus on genes shared across all mammals in our dataset and not newly functionalized genes in a lineage.

\begin{figure}[!ht]
\begin{adjustwidth}{-2.25in}{0in} % Comment out/remove adjustwidth environment if table fits in text column.
\centering
\includegraphics[width=1.4\textwidth, page=1] {figure2.eps}
\caption{
{\bf Selection coefficients at the phylogenetic and population scales.}
At the phylogenetic scale (A), we estimated the amino-acid fitness for each site from protein-coding DNA alignments using mutation-selection codon models.
For every possible mutation, the difference in amino-acid fitness before and after the mutation allows us to compute the selection coefficient at the phylogenetic scale ($\Sphy$).
Depending on $\Sphy$ (B) mutations can be predicted as deleterious ($\SphyDel$), nearly-neutral ($\SphyNeu$) or beneficial non-adaptive mutations ($\SphyBen$) toward a fitter amino acid and repairing existing functions.
At the population scale, each observed single nucleotide polymorphism (SNP) segregating in the population can also be classified according to its $\Sphy$ value (C).
Occurrence and frequency in the population of non-synonymous polymorphisms, contrasted to synonymous polymorphisms (deemed neutral) is used to estimate selection coefficients (D-E) at the population scale ($\Spop$), for each class of selection ($\SphyDel$, $\SphyNeu$, $\SphyBen$).
We can thus assess whether $\Sphy$ predicts $\Spop$ and compute \textit{precision} (F) and \textit{recall} (G) for each class.
The \textit{recall} value for class $\SphyBen$ is the probability for beneficial mutations to be non-adaptive (G).
Icons are adapted from \href{https://phylopic.org}{https://phylopic.org} under a Creative Commons license.
}
\label{fig:method}
\end{adjustwidth}
\end{figure}


Having identified which potential DNA changes represent beneficial non-adaptive mutations (Fig~\ref{fig:method}A-B), we retrieved polymorphism data from 28 wild and domesticated populations belonging to six genera (\textit{Equus}, \textit{Bos}, \textit{Capra}, \textit{Ovis}, \textit{Chlorocebus}, and \textit{Homo}) to assess the presence of beneficial non-adaptive mutations at the population scale.
We focused on both mutations currently segregating within populations and on substitutions in the terminal branches, and checked if any of these observed changes were indeed beneficial (Fig~\ref{fig:method}C-E).
A similar approach demonstrated the presence of beneficial non-adaptive mutations in humans~\cite{moses_inferring_2009, fischer_germline_2011} and in plants~\cite{chen_hunting_2021}.
However, the model used to reconstruct the static fitness landscape in these studies can only be applied to deeply conserved protein domains in the tree of life, which corresponds to a subpart of the proteome that evolves slowly.
The mutation-selection model used in the present work integrates phylogenetic relationships, and thus allows us to estimate the fitness landscape in shallower phylogenetic trees, and therefore can be applied almost exome-wide~\cite{rodrigue_mutationselection_2010}.

We first quantified the likelihood of any DNA mutation to be a beneficial non-adaptive mutation, that is, whenever a DNA mutation increases fitness under a stable fitness landscape.
Subsequently, by quantifying the total amount of beneficial mutations in the current population across all types of DNA mutations, we could tease apart beneficial non-adaptive from adaptive mutations resulting from a change in the fitness landscape.
Altogether, in this study, by integrating large-scale genomic datasets at both phylogenetic and population scales, we propose a way to explicitly quantify the contribution of beneficial non-adaptive mutations to positive selection across the entire exome of the six genera (figs.~\ref{fig:method}F-G).

\section*{Results}

\subsection*{Selection along the terminal branches}
First, we assessed whether fitness effects derived from the mutation-selection model at the phylogenetic scale predict selection occurring in terminal branches.
We recovered the mutations that reached fixation in the terminal branches of the six genera.
We only considered mutations fixed in a population as substitutions in the corresponding branch by discarding mutations segregating in our population samples.
For each substitution identified in the terminal branches we obtained its $\Sphy$ value such as predicted at the mammalian scale (Fig~\ref{fig:method}A-B).
We could classify each substitution as either deleterious $\SphyDel \coloneqq \Sphy < -1$), nearly-neutral ($\SphyNeu \coloneqq -1 < \Sphy <1$), or beneficial ($\SphyBen \coloneqq \Sphy > 1$)
Because $\Sphy$ values were based on the assumption that the fitness landscape is stable across mammals, $\SphyBen$ mutations (i.e., with $\Sphy>1$) bring the bearer of this mutation toward an amino acid predicted to be fitter across mammals.
Importantly, the mammalian alignment used to estimate the amino acid fitness landscape did not include the six focal genera and their sister species.
This ensures independence between, on the one hand, the fitness landscape estimated, and on the other hand, both substitutions that occurred in the terminal branches, and segregating polymorphisms of the focal populations.
Example substitutions in the terminal lineage of \textit{Chlorocebus sabaeus} which are classified as $\SphyBen$ are shown in supplementary materials (section S2).
For instance, in the mammalian protein-coding DNA alignment of gene \textit{SELE}, the nucleotide at site 1722 has mutated (from T to C) at the base of Simiiformes (monkeys and apes), modifying the corresponding amino acid from Serine to Proline, but has been subsequently reverted in the branch of \textit{Chlorocebus sabaeus} (\nameref{S2_Fig}).
However, other substitutions classified as $\SphyBen$ in the terminal branch of \textit{Chlorocebus sabaeus} cannot be clearly interpreted as reversions along the terminal branch, and show several transitions to this amino acid across the mammalian phylogeny, as for instance site 3145 of gene \textit{THSD7A} (\nameref{S3_Fig}).

Among all the substitutions found in each terminal branch, between 10 and 13\% were $\SphyBen$, while $\SphyBen$ mutations only represent between 0.9 and 1.2\% of all non-synonymous mutations (Fig~\ref{fig:homo-afr-results}A-B for humans and ~\nameref{S1_Table} for all dataset).
Of note, if we were to assume a stationary mutation-selection-drift equilibrium in the terminal lineage, we would expect a symmetric proportion of positively ($\SphyBen$) and negatively ($\SphyDel$) selected substitutions if there were no adaptation.
The lack of symmetry along the terminal branches then provides a means to estimate the frequency of non-adaptive beneficial substitutions.
Mathematically, twice the fraction of $\SphyBen$ substitutions is an estimate of this rate.
This rate is highly consistent across lineages (\nameref{S1_Table}) and suggests an overall frequency of nearly neutral substitutions due to consistent long-term selection pressures, mutation and drift of approximately 20\% of all substitutions (20\%-26\% across species).

\begin{figure}[!ht]
\begin{adjustwidth}{-2.25in}{0in} % Comment out/remove adjustwidth environment if table fits in text column.
\centering
\includegraphics[width=1.4\textwidth, page=1] {figure3.eps}
\caption{
{\bf Selection coefficients of mutations in humans of African descent.}
(A) Distribution of scaled selection coefficients ($\Sphy$), predicted for all possible non-synonymous DNA mutations away from the ancestral human exome (section \ref{subsec:nunber-of-sites}).
Mutations are divided into three classes of selection: deleterious ($\SphyDel$), nearly-neutral ($\SphyNeu$) and beneficial ($\SphyBen$, supposedly beneficial non-adaptive mutations)
(B) Distribution of scaled selection coefficients ($\Sphy$) for all observed substitutions along the \textit{Homo} branch after the \textit{Homo}-\textit{Pan} split (section \ref{subsec:substitution-mapping-in-the-terminal-branch}).
If there are fewer substitutions than expected, this class is thus undergoing purifying selection, as is the case for $\SphyDel$.
(C) The site-frequency spectrum (SFS) in humans of African descent for a random sample of 16 alleles (means in solid lines and standard deviations in color shades) for each class of selection and for synonymous mutations, supposedly neutral (black). The SFS represents the proportion of mutations (y-axis) with a given number of derived alleles in the population (x-axis).
At high frequencies, deleterious mutations are underrepresented.
(D) Proportion of beneficial $\ProbaPopDel$, nearly-neutral $\ProbaPopNeu$, and deleterious mutations $\ProbaPopBen$ estimated at the population scale for each class of selection at the phylogenetic scale (section \ref{subsec:s-polymorphism-method}). Proportions depicted here are not weighted by their mutational opportunities.
}
\label{fig:homo-afr-results}
\end{adjustwidth}
\end{figure}

Furthermore, since in principle, $\SphyBen$ mutations are bound to reach fixation more often than neutral mutations, we calculated the $\dnds$ ratio of non-synonymous over synonymous divergence for all terminal lineages, focusing on the non-synonymous changes predicted as $\SphyBen$ mutations ($\dn(\SphyBen) / \ds$).
We obtained values between 1.17 and 1.75 in the different lineages (\nameref{S2_Table}), meaning that $\SphyBen$ mutations reach fixation slightly more frequently than synonymous mutations that are supposed to be neutral, consistent with these $\SphyBen$ mutations being weakly beneficial.
Such an observation is consistent with the premise that $\SphyBen$ mutations are weakly beneficial, translating to a scaled selection coefficient between 0.32 and 1.24 (\nameref{S2_Table}).
Finding $\dn(\SphyBen) / \ds > 1$ for these sites confirms that these sites are closer to optimality at the end of the branch than at the beginning.
Even though the beneficial effect of these mutations does not come from an environmental change, it does not change the fact that they have contributed positively to the population’s fitness.
It is just that at mutation-selection-drift equilibrium, the increase in fitness at these sites is offset by deleterious substitutions elsewhere in the genome so that there is no net adaptation.

This result further indicates that using $\dnds$ as an estimate of purifying selection is biased (overestimated) due to the presence of beneficial non-adaptive mutations among the non-synonymous substitutions.
By discarding all beneficial non-adaptive mutations we can obtain an estimate of $\dnds$ which is not inflated.
By comparing these two ways of calculating $\dnds$ (see section~\ref{subsec:substitution-mapping-in-the-terminal-branch} in Materials \& Methods), we calculated that beneficial non-adaptive mutations inflate $\dnds$ values by between 9 and 12\% across genera.
This represents a substantial increase when considering that beneficial non-adaptive mutations only represent between 0.9 and 1.2\% of non-synonymous mutational opportunities (\nameref{S3_Table}).


\subsection*{Selection in populations}
Second, we assessed whether our calculated $\Sphy$ values predicted at the phylogenetic scale were also indicative of the selective forces exerted at the population level.
We retrieved single nucleotide polymorphisms (SNPs) segregating in 28 mammalian populations.
To determine if SNPs were ancestral of derived, we reconstructed the ancestral exome of each population.
We then classified every non-synonymous SNP as either $\SphyDel$, $\SphyNeu$, or $\SphyBen$ according its $\Sphy$ value (Fig~\ref{fig:method}B-C).

First, SNPs classified as $\SphyBen$ are spread across the genomes and not strongly associated to the ontology terms of their respective genes (\nameref{S4_Table},~\nameref{S4_Fig}).
In humans, some SNPs have been associated with specific clinical prognosis terms obtained by clinical evaluation of the impact of variants on human Mendelian disorders~\cite{landrum_clinvar_2018}.
Although this classification also relies on deep protein alignments and therefore cannot be considered an independent result from our own, it does provide a consistency check if the effect of a mutation on human health is in line with its fitness effect predicted by our method~\cite{grimm_evaluation_2015}.
Therefore, we investigated whether the non-synonymous SNPs classified as $\SphyDel$ or $\SphyBen$ showed enrichment in specific clinical terms compared to SNPs classified as $\SphyNeu$.
Our results show that SNPs predicted as deleterious are associated with clinical terms such as \textit{Likely Pathogenic} and \textit{Pathogenic}, implying that, in general, the selective pressure of a mutation exerted across mammals is also predictive of its clinical effect in humans (\nameref{S5_Table})~\cite{sullivan_leveraging_2023}.
Conversely, $\SphyBen$ mutations are associated with clinical terms such as \textit{Benign} and \textit{Likely Benign}, which shows that $\SphyBen$ mutations are less likely to be functionally damaging (\nameref{S6_Table}).

In addition to clinical prognosis, frequencies at which SNPs are segregating within populations provide information on their selective effects.
For instance, deleterious SNPs usually segregate at lower frequencies because of purifying selection, which tends to remove them from the population (Fig~\ref{fig:homo-afr-results}C for humans).
By gathering information across many SNPs, it is possible to estimate the distribution of fitness effects at the population scale, taking synonymous SNPs as a neutral expectation~\cite{eyre-walker_distribution_2006, eyre-walker_estimating_2009, galtier_adaptive_2016, tataru_inference_2017}.
From these estimated fitness effects, we can derive the proportion of deleterious mutations ($\ProbaPopDel$,~\nameref{S5_Fig}), nearly-neutral mutations ($\ProbaPopNeu$,~\nameref{S6_Fig}) and beneficial mutations ($\ProbaPopBen$,~\nameref{S7_Fig}) at the population scale (see section~\ref{subsec:s-polymorphism-method} in Materials \& Methods).
These approaches offer a unique opportunity to contrast selection coefficients estimated at the phylogenetic scale ($\Sphy$) and at the population scale ($\Spop$) in different dataset (\nameref{S8_Fig}).

Across our selection classes ($\SphyDel$, $\SphyNeu$ and $\SphyBen$), one can ultimately estimate the proportion of correct and incorrect predictions, leading to an estimation of \textit{precision} and \textit{recall} (Fig~\ref{fig:method}F-G and section~\ref{subsec:precisison_recall} in Materials \& Methods).
Across 28 populations of different mammal species, mutations predicted to be deleterious at the phylogenetic scale ($\SphyDel$) were indeed purged at the population scale, with a \textit{precision} in the range of 90--97\% (Table~\ref{table:proba} and Fig~\ref{fig:homo-afr-results}D for humans).
Conversely, a \textit{recall} in the range of 96--100\% implied that mutations found to be deleterious at the population scale were most likely also predicted to be deleterious at the phylogenetic scale (Table~\ref{table:proba}).
Altogether, purifying selection is largely predictable and amino acids with negative fitness across mammals have been effectively purged away in each population.

\begin{table}[!ht]
\begin{adjustwidth}{-2.25in}{0in} % Comment out/remove adjustwidth environment if table fits in text column.
\centering
\caption{
{\bf \textit{Precision} and \textit{recall} for estimated selection coefficient of mutations given by mutation-selection models ($\bm{\Sphy}$).}}
\scriptsize
\begin{tabular}{||l|l|r||r|r||r|r||r|r||}
\toprule
\multicolumn{3}{||c||}{} &
\multicolumn{2}{c||}{\shortstack{\textbf{Deleterious mutations} \\ $\bm{\SpopDel \coloneqq \Spop<-1}$ \\ $\bm{\SphyDel \coloneqq \Sphy<-1}$}} &
\multicolumn{2}{c||}{\shortstack{\textbf{Nearly-neutral mutations} \\ $\bm{\SpopNeu \coloneqq -1<\Spop<1}$ \\ $\bm{\SphyNeu \coloneqq -1<\Sphy<1}$}} &
\multicolumn{2}{c||}{\shortstack{\textbf{Beneficial mutations} \\ $\bm{\SpopBen \coloneqq \Spop>1}$ \\ $\bm{\SphyBen \coloneqq \Sphy>1}$}}
\\ \hline
\textbf{Population} &
\textbf{Species} &
$\bm{\Ne}$ &
\makecell{\textbf{Precision} \\ $\bm{\proba [\SpopDel \given \SphyDel]}$} &
\makecell{\textbf{Recall} \\ $\bm{\proba [\SphyDel \given \SpopDel]}$}            &
\makecell{\textbf{Precision}      \\ $\bm{\proba [\SpopNeu \given \SphyNeu]}$}                                    &
\makecell{\textbf{Recall}          \\ $\bm{\proba [\SphyNeu \given \SpopNeu]}$}                                  &
\makecell{\textbf{Precision}          \\ $\bm{\proba [\SpopBen \given \SphyBen]}$}          &
\makecell{\textbf{Recall}        \\ $\bm{\proba [\SphyBen \given \SpopBen]}$}
\\   \midrule
\rowcolor{LIGHTGREY} Equus c. & Equus caballus & $7.5\times 10^{4}$ & $ 0.923$ & $ 0.972$ & $ 0.570$ & $ 0.341$ & $ 0.648$ & $ 0.536$ \\
Iran & Bos taurus & $5.6\times 10^{4}$ & $ 0.915$ & $ 1.000$ & $ 0.632$ & $ 0.358$ & $ 0.873$ & $ 0.243$ \\
Uganda & Bos taurus & $1.3\times 10^{5}$ & $ 0.951$ & $ 0.969$ & $ 0.495$ & $ 0.414$ & $ 0.576$ & $ 0.415$ \\
\rowcolor{LIGHTGREY} Australia & Capra hircus & $1.7\times 10^{5}$ & $ 0.944$ & $ 0.971$ & $ 0.527$ & $ 0.437$ & $ 0.368$ & $ 0.177$ \\
\rowcolor{LIGHTGREY} France & Capra hircus & $1.9\times 10^{5}$ & $ 0.946$ & $ 0.971$ & $ 0.508$ & $ 0.423$ & $ 0.368$ & $ 0.190$ \\
\rowcolor{LIGHTGREY} Iran (C. aegagrus) & Capra hircus & $1.9\times 10^{5}$ & $ 0.948$ & $ 0.969$ & $ 0.486$ & $ 0.444$ & $ 0.368$ & $ 0.165$ \\
\rowcolor{LIGHTGREY} Iran & Capra hircus & $2.3\times 10^{5}$ & $ 0.953$ & $ 0.966$ & $ 0.425$ & $ 0.407$ & $ 0.368$ & $ 0.193$ \\
\rowcolor{LIGHTGREY} Italy & Capra hircus & $1.9\times 10^{5}$ & $ 0.947$ & $ 0.971$ & $ 0.551$ & $ 0.439$ & $ 0.368$ & $ 0.243$ \\
\rowcolor{LIGHTGREY} Morocco & Capra hircus & $2.2\times 10^{5}$ & $ 0.950$ & $ 0.970$ & $ 0.527$ & $ 0.440$ & $ 0.368$ & $ 0.245$ \\
Iran & Ovis aries & $3.8\times 10^{5}$ & $ 0.961$ & $ 0.961$ & $ 0.452$ & $ 0.415$ & $ 0.205$ & $ 0.407$ \\
Iran (O. orientalis) & Ovis aries & $4.5\times 10^{5}$ & $ 0.964$ & $ 0.960$ & $ 0.420$ & $ 0.445$ & $ 0.193$ & $ 0.190$ \\
Iran (O. vignei) & Ovis aries & $3.7\times 10^{5}$ & $ 0.967$ & $ 0.959$ & $ 0.361$ & $ 0.470$ & $ 0.190$ & $ 0.110$ \\
Various & Ovis aries & $4.1\times 10^{5}$ & $ 0.962$ & $ 0.962$ & $ 0.433$ & $ 0.440$ & $ 0.229$ & $ 0.222$ \\
Morocco & Ovis aries & $ 4\times 10^{5}$ & $ 0.962$ & $ 0.961$ & $ 0.462$ & $ 0.424$ & $ 0.211$ & $ 0.514$ \\
\rowcolor{LIGHTGREY} Barbados & Chlorocebus sabaeus & $1.1\times 10^{5}$ & $ 0.935$ & $ 0.975$ & $ 0.565$ & $ 0.402$ & $ 0.648$ & $ 0.293$ \\
\rowcolor{LIGHTGREY} Central Afr. Rep. & Chlorocebus sabaeus & $1.7\times 10^{5}$ & $ 0.948$ & $ 0.971$ & $ 0.508$ & $ 0.423$ & $ 0.535$ & $ 0.275$ \\
\rowcolor{LIGHTGREY} Ethiopia & Chlorocebus sabaeus & $1.4\times 10^{5}$ & $ 0.935$ & $ 0.975$ & $ 0.580$ & $ 0.416$ & $ 0.552$ & $ 0.245$ \\
\rowcolor{LIGHTGREY} Gambia & Chlorocebus sabaeus & $1.4\times 10^{5}$ & $ 0.944$ & $ 0.975$ & $ 0.654$ & $ 0.437$ & $ 0.577$ & $ 0.821$ \\
\rowcolor{LIGHTGREY} Kenya & Chlorocebus sabaeus & $1.5\times 10^{5}$ & $ 0.946$ & $ 0.972$ & $ 0.538$ & $ 0.453$ & $ 0.588$ & $ 0.257$ \\
\rowcolor{LIGHTGREY} Nevis & Chlorocebus sabaeus & $ 1\times 10^{5}$ & $ 0.933$ & $ 0.976$ & $ 0.629$ & $ 0.412$ & $ 0.599$ & $ 0.358$ \\
\rowcolor{LIGHTGREY} South Africa & Chlorocebus sabaeus & $1.8\times 10^{5}$ & $ 0.944$ & $ 0.971$ & $ 0.548$ & $ 0.423$ & $ 0.574$ & $ 0.341$ \\
\rowcolor{LIGHTGREY} Saint Kitts & Chlorocebus sabaeus & $1.2\times 10^{5}$ & $ 0.936$ & $ 0.975$ & $ 0.586$ & $ 0.402$ & $ 0.598$ & $ 0.336$ \\
\rowcolor{LIGHTGREY} Zambia & Chlorocebus sabaeus & $1.7\times 10^{5}$ & $ 0.945$ & $ 0.971$ & $ 0.512$ & $ 0.432$ & $ 0.585$ & $ 0.250$ \\
African & Homo sapiens & $5.6\times 10^{4}$ & $ 0.911$ & $ 0.976$ & $ 0.579$ & $ 0.325$ & $ 0.721$ & $ 0.349$ \\
Admixed American & Homo sapiens & $4.5\times 10^{4}$ & $ 0.902$ & $ 0.978$ & $ 0.584$ & $ 0.299$ & $ 0.690$ & $ 0.345$ \\
East Asian & Homo sapiens & $ 4\times 10^{4}$ & $ 0.905$ & $ 0.978$ & $ 0.585$ & $ 0.325$ & $ 0.688$ & $ 0.249$ \\
European & Homo sapiens & $4.2\times 10^{4}$ & $ 0.906$ & $ 0.978$ & $ 0.584$ & $ 0.329$ & $ 0.688$ & $ 0.248$ \\
South Asian & Homo sapiens & $4.4\times 10^{4}$ & $ 0.908$ & $ 0.978$ & $ 0.584$ & $ 0.342$ & $ 0.691$ & $ 0.224$ \\
\bottomrule
\end{tabular}
\normalsize
\begin{flushleft} \textit{Precision} is the estimation of the selection coefficient at population scale ($\Spop$) given that $\Sphy$ is known.
Conversely, \textit{recall} is the estimation of $\Sphy$ given selection coefficient at the population scale ($\Spop$) is known.
\textit{Recall} for beneficial mutations ($\proba [\SphyBen \given \SpopBen]$) is thus the proportion of beneficial non-adaptive mutations among all beneficial mutations.
$\Ne$ is the estimated effective population size for each population.
\end{flushleft}
\label{table:proba}
\end{adjustwidth}
\end{table}

Mutations predicted as $\SphyNeu$ were effectively composed of a mix of neutral and selected mutations with varying \textit{precision} (36--63\%) and \textit{recall} (32--45\%) across the different populations (Table~\ref{table:proba}, Fig~\ref{fig:homo-afr-results}D for humans).
The variable proportions between populations can be explained by the effective number of individuals in the population ($\Ne$), a major driver of selection efficacy.
Moreover, estimates of mutation rate per generation ($u$), from Bergeron \textit{et al.}~\cite{bergeron_evolution_2023} and Orlando \textit{et al.}~\cite{orlando_recalibrating_2013}, and Watterson’s $\theta$ obtained from the synonymous SFS as in Achaz~\cite{achaz_frequency_2009}, allow us to obtain $\Ne$ through $\Ne=\theta/4 u)$.
Using correlation analyses that accounted for phylogenetic relationship (see section~\ref{subsec:correlation-diversity} in Materials \& Methods,~\nameref{S9_Fig}), we found that higher $\Ne$ was associated with a smaller proportion of nearly-neutral mutations ($r^2=0.31$, $p=0.001$, Fig~\ref{fig:diversity}A).
This result follows the prediction of the nearly-neutral theory and suggests that in populations with higher diversity (e.g.,~\textit{Bos} or \textit{Ovis}), discrimination between beneficial and deleterious mutations is more likely to occur (\nameref{S10_Fig},~\nameref{S11_Fig},~\nameref{S12_Fig}).
Conversely, many more mutations are effectively neutral in populations with lower diversity (e.g.,~\textit{Homo}).

\begin{figure}[!ht]
\begin{adjustwidth}{-2.25in}{0in} % Comment out/remove adjustwidth environment if table fits in text column.
\centering
\includegraphics[width=1.4\textwidth, page=1] {figure4.eps}
\caption{
{\bf Proportion of nearly-neutral mutations and beneficial non-adaptive mutations as a function of effective population size ($\Ne$).}
Populations in circles, mean of the species across the populations as squares.
(A) Proportion of nearly-neutral mutations at the population scale ($\proba [ \SpopNeu]$ in the y-axis), shown as a function of estimated effective population size ($\Ne$ in the x-axis).
(B) Proportion of beneficial non-adaptive mutations among all beneficial mutations ($\proba [ \SphyBen \given \SpopBen]$ in the y-axis), shown as a function of $\Ne$ in x-axis.
Correlations account for phylogenetic relationship and non-independence of samples, through the fit of a Phylogenetic Generalized Linear Model (see section~\ref{subsec:s-polymorphism-method} in Materials \& Methods).
}
\label{fig:diversity}
\end{adjustwidth}
\end{figure}

Finally, mutations predicted to be $\SphyBen$ were indeed beneficial for individuals bearing them, with a \textit{precision} (Fig~\ref{fig:method}F) in the range of 19--87\% (Table~\ref{table:proba} and Fig~\ref{fig:homo-afr-results}D for humans).
This result confirms that selection toward amino acids restoring existing functions is ongoing in these populations.
Importantly, the \textit{recall} value in this case, computed as $\proba [ \SphyBen \given \SpopBen]$, is the probability for a beneficial mutation at the population scale to be a non-adaptive, i.e., going toward a fitter amino acid given a stable fitness landscape (Fig~\ref{fig:method}G,~\nameref{S7_Table}).
In other words, the \textit{recall} value quantifies the number of beneficial mutations restoring damaged genomes instead of creating adaptive innovations.
Across the 28 populations, this proportion is in the range of 11--82\% (Table~\ref{table:proba}), with a mean of 30\%.
Accounting for phylogenetic relationships, we found no correlation between the proportion of beneficial non-adaptive mutations and estimates of $\Ne$ based on genetic diversity ($r^2=0.00$, $p=0.772$, Fig~\ref{fig:diversity}B).

We additionally performed controls and simulations to ensure our results were robust.
First, we controlled that these estimations were not affected by SNP mispolarization (\nameref{S13_Fig},~\nameref{S14_Fig}).
Second, we performed simulations at the population-genetic level and confirmed that our method was able to recover the proportion of beneficial mutations that are non-adaptive in synthetic polymorphism datasets (\nameref{S15_Fig}).
Third, we ran our analysis filtering out CpG mutations and obtained values of $\proba [ \SphyBen \given \SpopBen]$ in the range of 5--27\%, with a mean of 14\% (\nameref{S8_Table}, \nameref{S9_Table}, \nameref{S16_Fig}), providing more conservative estimates.
Finally, because the phylogenetic mutation-selection codon model should fit better for genes with uniformly conserved functions, we filtered out genes under pervasive adaptation~\cite{latrille_genes_2023} as a control.
In this subset of the exome, containing genes with a more stable fitness landscape, we found an increase in the proportion of beneficial mutations that are non-adaptive (Wilcoxon signed-rank, $s=80$, $p=0.002$,~\nameref{S10_Table}), consistent with our expectation that beneficial mutations occur more frequently in genes under changing fitness landscapes.

\subsection*{Selection in the terminal lineage and in populations}

As an alternative to relying solely on currently segregating mutations to quantify selection, one can leverage both polymorphism within a population and substitutions in the terminal lineage to estimate the distribution of fitness effects (DFE).
Hence, we estimated \textit{precision} and \textit{recall} as done previously, but now including the number of substitutions per site as input for the DFE estimators (see~\ref{subsec:s-polymorphism-method} in Materials \& Methods and~\nameref{S17_Fig}).
When including substitutions in the terminal lineage, estimates of $\proba [ \SphyBen \given \SpopBen]$ are in the 10--78\% range with a mean of 36\%, and 19 out of 28 estimates fall between 15\% and 45\% (\nameref{S11_Table}, \nameref{S12_Table}).

Additionally, we controlled that these estimations were not affected by SNP mispolarization (\nameref{S18_Fig}).
We also filtered out genes under pervasive adaptation, and again found an increase in $\proba [ \SphyBen \given \SpopBen]$, consistent with our expectation (Wilcoxon signed-rank, $s=120$, $p=0.027$,~\nameref{S13_Table}).
We assessed the impact of fitting the same functional form of DFE to the three different categories of changes $\SphyDel$, $\SphyNeu$ and $\SphyBen$.
To this aim we computed the total amount of current selection by fitting either a single DFE on the whole dataset or by summing the other three independent DFEs.
These disjoint estimates are well correlated, with a goodness of fit $r^2={0.95, 0.89, 0.82}$ for respectively $\proba [\SpopDel]$ (\nameref{S19_Fig}), $\proba [\SpopNeu]$(\nameref{S20_Fig}) and $\proba [\SpopBen]$ (\nameref{S21_Fig}).
Finally, we evaluated the effect of fitting a parametric functional form for the DFE.
As implemented in Tataru \textit{et al.}~\cite{tataru_inference_2017}, the DFE is a mixture between a reflected gamma distribution and an exponential distribution (eq.~\ref{eq:parametric-dfe}, section~\ref{subsec:precisison_recall} in Materials \& Methods).
Instead of using such a continuous DFE, we also tested our prediction with a non-parametric functional form for the DFE, obtaining estimates of $\proba [ \SphyBen \given \SpopBen]$ in the 8--94\% range, with a mean of 43\% (\nameref{S14_Table}, \nameref{S15_Table}).

\section*{Discussion}
\subsection*{Beneficial mutations are not necessarily adaptive}

This study represents an essential step toward integrating the different evolutionary scales necessary to understand the combined effects of mutation, selection, and drift on genome evolution.
In particular, we have been able to quantify the proportion of beneficial mutations that are non-adapative (i.e., not a response to a change in fitness landscape), which has only been achievable by combining exome-wide data from both phylogenetic and population scales.
At the phylogenetic scale, codon diversity at each site of a protein-coding DNA alignment allows for reconstructing an amino-acid fitness landscape, assuming that this landscape is stable along the phylogenetic tree.
These amino-acid fitness landscapes allow us to predict any mutation’s selection coefficient ($\Sphy$) along a protein-coding sequence.
We have compared these selective effects to observations at the population level, and by doing so, we have confirmed that mutations predicted to be deleterious ($\SphyDel \coloneqq \Sphy < -1$) are generally purified away in extant populations.
Our results concur with previous studies showing that SIFT scores~\cite{ng_sift_2003, vaser_sift_2016}, based on amino acid alignments across species, also inform on the deleterious fitness effects exerted at the population scale~\cite{chen_hunting_2021}.
However, contrary to SIFT scores, our mutation-selection model is parameterized by a fitness function such that changes are directly interpretable as fitness effects (see also section S5).
In this regard, an interesting prediction of our model is that some deleterious mutations reach fixation due to genetic drift, while beneficial non-adaptive mutations restore states of higher fitness.
We have tested this hypothesis and have found that a substantial part of these predicted non-adaptive mutations ($\SphyBen \coloneqq \Sphy > 1 $) are indeed beneficial in extant populations.
We estimate that between 11 and 82\% of all beneficial mutations in mammalian populations are not adaptive.
More specifically, in 24 out of 28 populations analyzed, the percentage of beneficial mutations estimated to be non-adaptive falls between 15 and 45\%.
These results suggest that many beneficial mutations are not adaptive, but rather restore states of higher fitness.
Hence, we can correctly estimate the extent of adaptive evolution only if we account for the number of beneficial non-adaptive mutations~\cite{keightley_what_2010, rice_evolutionarily_2015}.
Here instead, we argue that we should dissociate positive selection from adaptive evolution and limit the use of adaptive mutations to those that are associated with adaptation to environmental change as such~\cite{charlesworth_other_2007, mustonen_fitness_2009, mustonen_fitness_2010}.

\subsection*{Assumptions and methodological limitations}

The exact estimation of the contribution of beneficial non-adaptive mutations to positive selection relies on some hypotheses at both the phylogenetic and population scales and is sensitive to methodological limitations.
Indeed, data quality and potentially inadequate modeling choices of both the fitness landscape (at the phylogenetic scale) and fitness effects (at the population scale) might also lead to missed predictions~\cite{moutinho_variation_2019}.
In practice, we obtained different values of the proportion of non-adaptive beneficial mutations depending on i) the filtering or not of CpG mutations~\cite{eyre-walker_how_2014}, ii) whether we included substitutions in the terminal lineage along with within-population polymorphisms to estimate fitness effects~\cite{tataru_inference_2017}, and iii) the model used to infer the fitness effects.
It appears that our estimation can be sensitive to model misspecification and overall, while we provide an order of magnitude for the contribution of beneficial non-adaptive mutations to positive selection, methodological improvement on the estimation of the DFE is needed to increase the precision of this value.

To be conservative, we considered mutations as adaptive if they were detected as being under positive selection at the population scale despite them being either incorrectly predicted as deleterious ($\SphyDel$) or nearly-neutral ($\SphyNeu \coloneqq -1 < \Sphy <1$) from the amino-acid mammalian fitnesses.
An example of an incorrectly predicted deleterious mutation ($\SphyDel$) from its fitness landscape could be an amino acid having always been deleterious across mammals, but being advantageous ($\SpopBen$) in the current species due to environmental changes or a major shift in their fitness landscape (e.g. domestication)
To visualize an example of a wrongly predicted nearly-neutral mutation, we can first imagine a site where only hydrophilic amino acids are accepted because of the protein properties (e.g. a surface site of a globular protein).
Let us then assume that such a site is also a target for viruses, hence promoting amino-acid changes which modify the site's viral affinity~\cite{enard_viruses_2016}.
Given the selective pressure favoring amino-acid change, but restricting the possibilities to hydrophilic amino acids, most hydrophilic amino acids will likely be visited along the phylogenetic tree and the mutation-selection model will give high and similar fitnesses to all of them.
In such a case, any mutation between hydrophilic amino acids will be wrongly predicted as nearly-neutral ($\SphyNeu$), while it is in fact adaptive.
In summary, under a changing fitness landscapes~\cite{mustonen_molecular_2008}, our phylogenetic mutation-selection model takes an average over fitness changes observed along the phylogeny, causing beneficial mutations ($\SpopBen$) to be predicted as either deleterious ($\SphyDel$) or nearly-neutral ($\SphyNeu$), therefore mechanically reducing $\proba [ \SphyBen \given \SpopBen]$, and making our estimate conservative.

\subsection*{Convergent adaptation}

If there are several substitutions toward the same amino acid along the mammalian tree (\nameref{S3_Fig}), our mutation-selection model cannot formally distinguish between a scenario where mammals have fixed deleterious mutations that are reverted in several lineages, from more complex scenarios involving convergent adaptation across mammals.
In a first scenario, repeated changes of fitness landscapes in the same direction could occur along several lineages, leading to repeated substitutions in multiple lineages (parallel or convergent adaptation).
In a second scenario, an environmental change that occurred near the root of placental mammals ($\sim$100 Mys ago), to which extant populations are currently responding independently through weakly adaptive mutations, could also lead to repeated substitutions toward the same amino acids.
Importantly, we would usually expect adaptive convergent mutations to be linked to particular converging phenotypes across mammals, and hence, they should not massively affect the whole genome as we find (\nameref{S4_Table},~\nameref{S4_Fig}).
Moreover, after filtering out genes usually associated to recurrent adaptation (e.g. immune genes), we recover an even higher proportion of beneficial non-adaptive mutations (\nameref{S8_Table}).
For these reasons, we argue that the signal of predictable positive selection we recover in extant population is indeed mainly driven by non-adaptive evolution.

\subsection*{The influence of effective population size}

Across the genome, beneficial non-adaptive mutations and deleterious mutations reaching fixation create a balance in which genomes are constantly damaged and restored simultaneously at different loci due to drift.
Since the probability of fixation of mutations depends on the effective population size ($\Ne$), the history of $\Ne$ plays a crucial role in determining the number of beneficial non-adaptive mutations compensating for deleterious mutations~\cite{latrille_inferring_2021}.
For example, a population size expansion will increase the efficacy of selection, and a larger proportion of mutations will be beneficial (otherwise effectively neutral), thus increasing the number of beneficial non-adaptive mutations.
On the other hand, a population that has experienced a high $\Ne$ throughout its history should be closer to an optimal state under a stable fitness landscape, having suffered fewer fixations of deleterious mutations and therefore decreasing the probability of beneficial non-adaptive mutations~\cite{huber_determining_2017}.
Overall, we expect the proportion of beneficial non-adaptive mutations to be more dependent on $\Ne$’s long-term expansions and contractions than on the short-term ones~\cite{charlesworth_other_2007,huber_determining_2017}.

Moreover, because our model assumes a fixed fitness landscape, it implicitly assumes that $\Ne$ is constant along the phylogenetic tree.
Fluctuations due to changes in the fitness landscape or in $\Ne$ will be averaged out by the assumption of the current model that $\Ne$ is constant across lineages.
It was recently shown~\cite{latrille_inferring_2021}, using computer intensive mutation-selection models with fluctuating $\Ne$, that relaxing the assumption of a constant $\Ne$ results in more extreme estimates of amino-acid fitnesses than with the standard model used in this study.
In other words, by assuming a constant $\Ne$, we are underpowered to detect beneficial non-adaptive mutations since amino acids will have more similar fitnesses.
As a consequence, some of the beneficial non-adaptive mutations currently segregating in population will be incorrectly classified as nearly-neutral by the mutation-selection model, and thus be wrongly interpreted as adaptive (see previous section).
This ultimately results in lower estimates of the proportion of beneficial non-adaptive mutations.
Given this inflation of missed predictions due to change in population sizes~\cite{lanfear_population_2014, jones_shifting_2017, platt_protein_2018}, our estimated proportion of beneficial non-adaptive mutations among adaptive ones is likely to be an underestimation.

\subsection*{The role of epistasis and compensatory mutations}

Our model assumes that amino-acid fitness landscapes are site-specific and also independent of one another, whereas under pervasive epistasis, the fitness effect of any mutation at a particular site would depend on the amino acids present at other sites.
Epistasis is common for mutations that influence the protein’s physical properties (e.g. conformation, stability, or affinity for ligands) or might arise due to nonlinear relationship between the protein’s physical properties and fitness~\cite{starr_epistasis_2016}.
Regardless of its origin, epistasis has been shown to play a role in the evolution of protein-coding genes, with amino-acid residues in contact within a protein or between proteins tending to co-evolve~\cite{morcos_directcoupling_2011, marks_protein_2012, starr_epistasis_2016}.
Particularly, the residues in contact co-evolve to become more compatible with each other generating an entrenchment~\cite{goldstein_nonadaptive_2015, goldstein_sequence_2017, park_epistatic_2022}.
Epistasis therefore allows for compensatory mutations, which restore fitness through mutations at loci different from where deleterious mutations took place, representing another case of non-adaptive beneficial mutations, but one which is not accounted for by our method.
Hence, the beneficial mutations that we classify as putatively adaptive might in fact be compensatory mutations, making our estimation of the rate of non-adaptive beneficial mutations conservative.

Despite epistasis being an important factor in protein evolution, several deep-mutational scanning experiments have revealed that a site-specific fitness landscape predicts the evolution of sequences in nature with considerable accuracy~\cite{ashenberg_mutational_2013, doud_sitespecific_2015, bloom_identification_2017}.
Additionally, the fact that we observe such a high proportion of beneficial non-adaptive mutations suggests that the underlying assumptions of our model, namely site-independence, implying no epistasis, and a static fitness landscape, are a reasonable approximation for the underlying fitness landscape of proteins.
Our results imply that the fitness effects of new mutations are mostly conserved across mammalian orthologs, in agreement with other studies showing that for conserved orthologs with similar structures and functions, models without epistasis provide a reasonable estimate of fitness effects in protein coding genes~\cite{youssef_consequences_2020, vigue_deciphering_2022}.
Conceptually, the framework presented here, with the addition of a more complex protein fitness landscape at the phylogenetic scale, could be used to infer the relative contribution of compensatory mutations to non-adaptive and adaptive evolution.

\subsection*{Detecting adaptation above the nearly-neutral background}

A long-standing debate in molecular evolution is whether the variation we observe between species in protein-coding genes is primarily due to nearly-neutral mutations reaching fixation by drift or primarily due to adaptation~\cite{kimura_evolutionary_1968,ohta_nearly_1992,gillespie_substitution_1994,jensen_importance_2019}.
Measuring the ``rate of adaptation'' in proteins, as pioneered by McDonald \& Kreitman~\cite{mcdonald_adaptative_1991}, has been central to inform this debate~\cite{galtier_half_2024}.
However, the McDonald \& Kreitman test detects signatures of accelerated evolution in a given terminal branch compared to an expectation based on polymorphism present in the population. It considers the fraction of substitutions that fix too quickly as ``adaptive''~\cite{mcdonald_adaptative_1991, smith_adaptive_2002, welch_estimating_2006} despite there being other processes that can lead to their fixation~\cite{galtier_adaptation_2007,rousselle_influence_2019,joseph_increased_2024a} and some of these substitutions being beneficial but non-adaptive~\cite{hartl_compensatory_1996, charlesworth_other_2007, mustonen_fitness_2009, jones_shifting_2017}.
Here, the expectation is built on the pattern of substitutions across a phylogeny compared to the fitness effects that can be estimated from both substitutions in a terminal lineage and polymorphism in populations.
Moreover, the goal is not to detect a fraction of beneficial substitution (i.e. ``adaptive'' substitutions for McDonald \& Kreitman~\cite{mcdonald_adaptative_1991}), but to estimate the proportion of non-adaptive mutations among beneficial ones.

We provide evidence that in mammalian orthologs, many substitutions occur through fixation of both deleterious mutations and beneficial non-adaptive mutations.
Detecting adaptation above this background of substitutions remains a challenge~\cite{kimura_evolutionary_1968, ohta_development_1996}.
Mathematically, the surplus of positive selection due to an externally-driven changing fitness landscape is called \textit{fitness flux}, and requires experimentally measuring the selection coefficient of each mutation in each genetic background.
The \textit{fitness flux} can be estimated if either the substitutions history is known~\cite{mustonen_fitness_2009} or changes of frequency in currently segregating variants~\cite{mustonen_fitness_2010}.
Without experimentally measured selection coefficients, another strategy is precisely to use a nearly-neutral substitution model as a null model of evolution.
Under a strictly neutral evolution of protein-coding sequence, we expect the ratio of non-synonymous over synonymous substitutions ($\dnds$) to be equal to one.
Deviations from this neutral expectation, such as $\dnds > 1$, which can be generated by an excess of non-synonymous substitutions, is generally interpreted as a sign of adaptation.
However, as shown in this study, a $\dnds > 1$ is not necessarily a signature of adaptation but can be due to beneficial non-adaptive mutations.
So, by relaxing the strict neutrality and assuming a stable fitness landscape instead, one can predict the expected rate of evolution, called $\omega_0$~\cite{spielman_relationship_2015, dosreis_how_2015}.
Adaptation can thus be considered as evolution under a changing fitness landscape and tested as such by searching for the signature of $\dnds > \omega_0$~\cite{cvijovic_fate_2015, rodrigue_detecting_2017, rodrigue_bayesian_2021}.
Using a stable fitness landscape as a null model of evolution, thus accounting for selective constraints exerted on the different amino acids, increased the statistical power in testing for adaptation~\cite{latrille_genes_2023}.
Instead of relying solely on summary statistics (such as $\dnds$ or $\omega_0$), another strategy to detect adaptation is to include changes in the fitness landscapes inherently within the mutation selection framework, either with small changes along the phylogeny ~\cite{tamuri_mutationselection_2021} or either by allowing fitness to change on subsets of branches ~\cite{kazmi_detecting_2019, stolyarova_senescence_2020}.
Such mechanistic models could be more general than site-specific fitness landscapes, including epistasis and changing fitness landscapes~\cite{goldstein_sequence_2017, stolyarova_senescence_2020}.

\subsection*{Conclusions}

We have provided empirical evidence that an evolutionary model assuming a stable fitness landscape at the mammalian scale allows us to predict the fitness effects of mutations in extant populations and individuals, acknowledging the balance between deleterious and beneficial non-adaptive mutations.
We argue that such a model would represent a null expectation for the evolution of protein-coding genes in the absence of adaptation.
Altogether, because a substantial part of positive selection can be explained by beneficial non-adaptive mutations, but not its entirety, we argue that the mammalian exome is shaped by both adaptive and non-adaptive processes, and that none of them alone is sufficient to explain the observed patterns of changes.
In that sense, to avoid conflating beneficial mutations with adaptive evolution, the term ``adaptation`` should retain its original meaning associated with a change in the underlying fitness landscape and be modelled as such~\cite{mustonen_fitness_2009, mustonen_fitness_2010}.

\newpage
\section{Materials \& Methods}
\label{sec:methods}

\subsection{Phylogenetic dataset}\label{subsec:phylo-dataset}

Protein-coding DNA sequence alignments in placental mammals and their corresponding gene trees come from the \href{https://www.orthomam.univ-montp2.fr}{OrthoMaM} database and were processed as in Latrille \textit{et al.}~\cite{latrille_genes_2023}.
OrthoMaM contains a total of $116$ mammalian reference sequences in v10c~\cite{ranwez_orthomam_2007, douzery_orthomam_2014, scornavacca_orthomam_2019}.

Genes located on the X and Y chromosomes and on the mitochondrial genome were discarded from the analysis because the level of polymorphism –~which is necessary for population-based analyses~– is expected to be different in these three regions compared to the autosomal genome.
Sequences of species for which we used population-level polymorphism (see section~\ref{subsec:polymorphism-dataset}) and their sister species, were removed from the analysis to ensure independence between the data used in the phylogenetic and population scales.
Sites in the alignment containing more than 10\% of gaps across the species were discarded.
Altogether, our genome-wide dataset contains $14,509$ protein-coding DNA sequences in $87$ placental mammals.

\subsection{Selection coefficient (\texorpdfstring{$\bm{\Sphy}$}{S₀}) in a phylogeny-based method}
\label{subsec:s-phylogeny-method}

We analyzed the phylogenetic-level data using mutation-selection models.
These models assume the protein-coding sequences are at mutation-selection balance under a fixed fitness landscape characterized by a fitness vector over the $20$ amino acids at each site~\cite{yang_mutationselection_2008, halpern_evolutionary_1998, rodrigue_mechanistic_2010}.
Mathematically, the rate of non-synonymous substitution from codon $a$ to codon $b$ ($q_{a \mapsto b}^{(i)}$) at site $i$ of the sequence is equal to the rate of mutation of the underlying nucleotide change ($\mu_{a \mapsto b}$) multiplied by the scaled probability of mutation fixation ($\proba_{a \mapsto b}^{(i)}$).
The probability of fixation depends on the difference between the scaled fitness of the amino acid encoded by the mutated codon ($F_b^{(i)}$) and the amino acid encoded by the original codon ($F_a^{(i)}$) at site $i$~\cite{wright_evolution_1931, fisher_genetical_1930}.

The rate of substitution from codon $a$ to $b$ at a site $i$ is thus:
\begin{equation}
\begin{dcases}
q_{a \mapsto b}^{(i)} & = 0 \text{ if codons $a$ and $b$ are more than one mutation away,} \\
q_{a \mapsto b}^{(i)} & = \mu_{a \mapsto b} \text{ if codons $a$ and $b$ are synonymous, and}\\
q_{a \mapsto b}^{(i)} & = \mu_{a \mapsto b} \dfrac{F_b^{(i)} - F_a^{(i)}}{1 - \e^{F_a^{(i)} - F_b^{(i)}}} \text{ if codons $a$ and $b$ are non-synonymous}.
\end{dcases}
\end{equation}

Fitting the mutation-selection model on a multi-species sequence alignment leads to an estimation of the gene-wide $4 \times 4$ nucleotide mutation rate matrix ($\bm{\mu}$) as well as the $20$ amino-acid fitness landscape ($\bm{F^{(i)}}$) at each site $i$.
The priors and full configuration of the model are given in section S1.
From a technical perspective, the Bayesian estimation is a two-step procedure~\cite{rodrigue_bayesian_2008}.
The first step is a data augmentation of the alignment, consisting in sampling a detailed substitution history along the phylogenetic tree for each site, given the current value of the model parameters.
In the second step, the parameters of the model can then be directly updated by a Gibbs sampling procedure, conditional on the current substitution history.
Alternating between these two sampling steps yields a Markov chain Monte-Carlo (MCMC) procedure whose equilibrium distribution is the posterior probability density of interest~\cite{lartillot_bayesian_2004, rodrigue_bayesian_2008}.
Additionally, across-site heterogeneities in amino-acid fitness profiles are captured by a Dirichlet process.
More precisely, the number of amino-acid fitness profiles estimated is lower than the number of sites in the alignment.
Consequently each profile has several sites assigned to it, resulting in a particular configuration of the Dirichlet process.
Conversely, sites with similar signatures are assigned to the same fitness profile.
This configuration of the Dirichlet process is resampled through the MCMC to estimate a posterior distribution of amino acid profiles for each site specifically~\cite{rodrigue_mutationselection_2010, lartillot_inference_2013}.
From a more mechanistic perspective, even though not all amino acids occur at every single codon site of the DNA alignment, we can nevertheless estimate the distribution of amino-acid fitnesses by generalizing the information recovered across sites and across amino acids based on the phylogenetic relationship among samples.
In particular, synonymous substitutions along the tree contain the signal to estimate branch lengths and the nucleotide transition matrix, while non-synonymous substitutions contain information on fitness difference between codons connected by single nucleotide changes~\cite{rodrigue_mutationselection_2010}.

The selection coefficient for a mutation from codon $a$ to codon $b$ at site $i$ is defined as:
\begin{equation}
\Sphy^{(i)} (a \mapsto b) = \Delta F^{(i)} = F^{(i)}_{b} - F^{(i)}_{a}.
\end{equation}
In our subsequent derivation the source ($a$) and target ($b$) codons as well as the site ($i$) are implicit and thus never explicitly written.

The scaled selection coefficient ($\Sphy = \Delta F$) is formally the product of the selection coefficient at the individual level ($s$) and the effective population size ($\Ne$), as $\Sphy = 4 \Ne \times s$.
The value of $\Sphy$ informs us on the strength of selection exerted on amino acids changes.
Thus, according to its $\Sphy$ value, we can classify any mutation as either a deleterious mutation toward a less fit amino acid ($\SphyDel \coloneqq \Sphy < -1$), a nearly-neutral mutation ($\SphyNeu \coloneqq -1 < \Sphy <1$), or a mutation toward a known fitter amino acid, constituting thus a beneficial non-adaptive mutation ($\SphyBen \coloneqq \Sphy > 1$).

We used the Bayesian software \textit{BayesCode} (\url{https://github.com/ThibaultLatrille/bayescode}, v1.3.1) to estimate the selection coefficients for each protein-coding gene in the mammalian dataset.
We ran the MCMC algorithm implemented in BayesCode for $2,000$ generations as described in Latrille \textit{et al.}~\cite{latrille_genes_2023}.
For each gene, after discarding a burn-in period of $1,000$ generations of MCMC, we obtained posterior mean estimates (over the $1,000$ generations left of MCMC) of the mutation rate matrix ($\bm{\mu}$) as well as the $20$ amino-acid fitness landscape ($\bm{F^{(i)}}$) at each site $i$.

\subsection{Polymorphism dataset}
\label{subsec:polymorphism-dataset}

The genetic variants representing the population level polymorphisms were obtained from the following species and their available datasets: \textit{Equus caballus} (EquCab2 assembly in the EVA study PRJEB9799~\cite{alabri_whole_2020}), \textit{Bos taurus} (UMD3.1 assembly in the NextGen project: \url{https://projects.ensembl.org/nextgen/}), \textit{Ovis aries} (Oar\_v3.1 assembly in the NextGen project), \textit{Capra hircus} (CHIR1 assembly in the NextGen project), converted to ARS1 assembly with dbSNP identifiers~\cite{sherry_dbsnp_2001}), \textit{Chlorocebus sabaeus} (ChlSab1.1 assembly in the EVA project PRJEB22989~\cite{svardal_ancient_2017}), \textit{Homo sapiens} (GRCh38 assembly in the 1000 Genomes Project~\cite{zheng-bradley_alignment_2017}).
In total, we analyzed 28 populations across the 6 different species with polymorphism data.
The data was processed as described in Latrille \textit{et al.}~\cite{latrille_genes_2023}.

Only bi-allelic single nucleotide polymorphisms (SNPs) found within a gene were in our polymorphism dataset, while nonsense variants and indels were discarded.
To construct the dataset, we first recovered the location of each SNP (represented by its chromosome, position, and strand) in the focal species and matched it to its corresponding position in the coding sequence (CDS) using gene annotation files (GTF format) downloaded from Ensembl (\url{ensembl.org}).
We then verified that the SNP downloaded from Ensembl matched the reference in the CDS in FASTA format.
Next, the position in the CDS was converted to the corresponding position in the multi-species sequence alignment (containing gaps) from the OrthoMaM database (see section~\ref{subsec:s-phylogeny-method}) for the corresponding gene by doing a global pairwise alignment (Biopython function pairwise2).
This conversion from genomic position to alignment position was only possible when the assembly used for SNP-calling was the same as the one used in the OrthoMaM alignment, the GTF annotations, and the FASTA sequences.
SNPs were polarized using the three closest outgroups found in the OrthoMaM alignment with est-usfs v2.04~\cite{keightley_inferring_2018}, and alleles with a probability of being derived lower than 0.99 were discarded.

\subsection{Mutational opportunities}
\label{subsec:nunber-of-sites}
The mutational opportunities of any new mutation refer to its likelihood of falling into a specific category (synonymous, deleterious, nearly-neutral, or beneficial).
Deriving such opportunities is necessary to estimate the strength of selection exerted at the population scale since different categories might have different mutational opportunities, and thus polymorphism and divergence need to be corrected accordingly (see sections~\ref{subsec:substitution-mapping-in-the-terminal-branch},~\ref{subsec:s-polymorphism-method}, and~\ref{subsec:precisison_recall}).
To calculate mutational opportunities, we reconstructed the ancestral exome of each of the 28 populations by using the most likely ancestral state from est-usfs (see section~\ref{subsec:polymorphism-dataset}), which differs from the corresponding species reference exome since it accounts for the variability present in the specific population.

From the reconstructed ancestral exome, all possible mutations were computed, weighted by the instantaneous rate of change between nucleotides obtained from the mutation rate matrix ($\bm{\mu}$, see section~\ref{subsec:s-phylogeny-method}), summing to $\mu_{\text{tot}}$ across the whole exome, and to $\mu_{\text{syn}}$ when restricted to synonymous mutations.
Finally, the mutational opportunities for synonymous mutations were computed as the total number of sites across the exome ($L_{\text{tot}}$) weighted by the proportion of synonymous mutations among all possible mutations as:

\begin{equation}
L_{\text{syn}} = L_{\text{tot}} \frac{\mu_{\text{syn}}}{\mu_{\text{tot}}}.
\end{equation}

Similarly, for non-synonymous mutations, the total mutation rate for each class of selection $\Sphyclass \in \{\SphyDel, \SphyNeu, \SphyBen \}$, called $\mu\left( \Sphyclass \right)$, was estimated as the sum across all non-synonymous mutations if their selection coefficient at the phylogenetic scale is in the class $\Sphy \in \Sphyclass$.
Accordingly, the mutational opportunities ($L \left( \Sphyclass \right)$) for each class of selection coefficient ($\Sphyclass$) was finally computed as the total number of sites across the exome ($L_{\text{tot}}$) weighted by the ratio of the aggregated mutations rates falling in the class $\mu\left( \Sphyclass \right)$:
\begin{equation}
L \left( \Sphyclass \right) = L_{\text{tot}} \frac{\mu\left( \Sphyclass \right)}{\mu_{\text{tot}}}.\label{eq:mutation-opp}
\end{equation}

Finally, $\proba [ \Sphyclass ]$ is the probability for a non-synonymous mutation to be in the class $\Sphyclass$, thus computed as:
\begin{equation}
\proba[\Sphyclass] = \frac{L\left( \Sphyclass \right)}{\sum_{\SphyclassAlt\in \{\SphyDel, \SphyNeu, \SphyBen \} } L\left(\SphyclassAlt \right)}.\label{eq:proba-dfe-mutsel}
\end{equation}

\subsection{Substitution mapping and \texorpdfstring{$\bm{\dnds}$}{dₙ/dₛ} in the terminal branch}
\label{subsec:substitution-mapping-in-the-terminal-branch}
We inferred the protein-coding DNA sequences for each node of the 4-taxa tree containing the focal species and the three closest outgroups species found in the OrthoMaM alignment by applying the M5 codon model (gamma site rate variation) as implemented in FastML.v3.11~\cite{ashkenazy_fastml_2012}.
Consequently, for each focal species we reconstructed the protein coding DNA sequence of the whole exome at the base of the terminal branch before the split from the sister species.
We considered \textit{Ceratotherium simum simum} as \textit{Equus caballus'} sister species; \textit{Bison bison bison} as \textit{Bos taurus'} sister species; \textit{Pantholops hodgsonii} as \textit{Ovis aries'} sister species; \textit{Pantholops hodgsonii} as \textit{Capra hircus'} sister species; \textit{Macaca mulatta} as \textit{Chlorocebus sabaeus'} sister species and finally, we considered \textit{Pan troglodytes} as \textit{Homo sapiens'} sister species.
From this reconstructed exome, we determined the direction of the substitution occurring along the terminal branch of the phylogenetic tree toward each extant population.
SNPs segregating in the population were discarded, and the most likely ancestral state from est-usfs (see section~\ref{subsec:polymorphism-dataset}) was used as the reference for each extant population.
For each substitution, we recovered its $\Sphy$ value as calculated through the phylogeny-based method (see section~\ref{subsec:s-phylogeny-method}).
Finally, the rate of non-synonymous over synonymous substitutions for a given class of selection coefficient ($\Sphyclass \in \{\SphyDel, \SphyNeu, \SphyBen \}$) was computed as:
\begin{align}
\begin{dcases}
\dn \left( \Sphyclass \right) &= \dfrac{D\left( \Sphyclass \right)}{L \left( \Sphyclass \right)}, \\
\ds &= \dfrac{D_{\text{syn}}}{L_{\text{syn}}},
\end{dcases}\label{eq:dnds}
\end{align}
where $D \left( \Sphyclass \right) $ was the number of non-synonymous substitutions in class $\Sphyclass$, $D_{\text{syn}}$ was the number of synonymous substitutions across the exome, while $L \left( \Sphyclass \right)$ and $L_{\text{syn}}$ were the numbers of non-synonymous and synonymous mutational opportunities, respectively, as defined in section~\ref{subsec:nunber-of-sites}.
$\delta(\dnds)$ was computed as the difference between $\dnds$ computed over all substitutions and $\dnds$ when we removed beneficial non-adaptive mutations $\dn (\Sphy < 1) / \ds$, normalized by $\dnds$.
Note that the quantities $\delta(\dnds)$ and $\delta(\dn)$ are equivalent due to the simplification of the factor $\ds$:
\begin{equation}
\delta(\dnds) = \dfrac{\dnds - \dn(\Sphy < 1) / \ds}{\dnds} = \dfrac{\dn - \dn(\Sphy < 1)}{\dn} = \delta(\dn).\label{eq:delta-dnds}
\end{equation}

\subsection{Scaled selection coefficients (\texorpdfstring{$\bm{\Spop}$}{S}) in a population-based method}
\label{subsec:s-polymorphism-method}

To obtain a quantitative estimate of the distribution of selection coefficients for each category of SNPs, we used the \textit{polyDFE} model~\cite{tataru_inference_2017, tataru_polydfe_2020}.
This model uses the count of derived alleles to infer the distribution of fitness effects (DFE).
The probability of sampling an allele at a given frequency (before fixation or extinction) is informative of its scaled selection coefficient at the population scale ($\Spop$).
Therefore, pooled across many sites, the site-frequency spectrum (SFS) provides information on the underlying $\Spop$ of mutations.
However, estimating a single $\Spop$ for all sampled mutations is biologically unrealistic, and a DFE of mutations is usually assumed~\cite{eyre-walker_distribution_2006, eyre-walker_estimating_2009}.
The \textit{polyDFE}\cite{tataru_inference_2017, tataru_polydfe_2020} software implements a mixture of a $\Gamma$ and exponential distributions to model the DFE of non-synonymous mutations, while synonymous mutations are considered neutral.
The model estimates the parameters $\DelMean$, $b$, $p_b$ and $\AdvMean$ for non-synonymous mutations as:

\begin{equation}
\label{eq:parametric-dfe}
\phi \left( \Spop; \DelMean , b, p_b, \AdvMean \right) =
\begin{dcases}
\left( 1 - p_b \right) f_{\Gamma}(-\Spop; -\DelMean, b) & \text{ if $\Spop \leq 0$,} \\
p_b f_{e}(\Spop; \AdvMean) & \text{ if $\Spop > 0$,} \\
\end{dcases}
\end{equation}

where $\DelMean \leq -1 $ is the estimated mean of the DFE for $\Spop \leq 0$;
$b \geq 0.2$ is the estimated shape of the $\Gamma$ distribution;
$0 \leq p_b \leq 1$ is the estimated probability that $\Spop > 0$;
$\AdvMean \geq 1$ is the estimated mean of the DFE for $\Spop > 0$;
and $f_{\Gamma}(\Spop; m, b)$ is the density of the $\Gamma$ distribution with mean $m$ and shape $b$, while $f_{e}(\Spop; m)$ is the density of the exponential distribution with mean $m$.

\textit{PolyDFE} requires one SFS for non-synonymous mutations and one for synonymous mutations (neutral expectation), as well as the number of sites on which each SFS was sampled.
For populations containing more than $8$ individuals, the SFS was subsampled down to $16$ chromosomes ($8$ diploid individuals) without replacement (hyper-geometric distribution) to alleviate the effect of different sampling depths in the 28 populations.
Altogether, for each class of selection ($\Sphyclass \in \{\SphyDel, \SphyNeu, \SphyBen \}$) of non-synonymous SNPs, we aggregated all the SNPs in the selection class $\Sphyclass$ as an SFS.
The number of sites on which each SFS was sampled is given by $L(\Sphyclass)$ for the non-synonymous SFS and $L_{\text{syn}}$ for the synonymous SFS respectively.
For each class of selection $\Sphyclass$, once fitted to the data using maximum likelihood with \textit{polyDFE}, the parameters of the DFE $\left( \DelMean , b, p_b, \AdvMean \right)$ were used to compute $\proba [ \SpopDel \given  \Sphyclass] $, $\proba [ \SpopNeu \given \Sphyclass]$, and $\proba [ \SpopBen \given \Sphyclass]$ as:

\begin{align}
\proba [ \SpopDel \given  \Sphyclass] &= \proba [ \Spop < -1 \given \Sphyclass ] = \left( 1 - p_b \right) \int_{-\infty}^{-1} f_{\Gamma}(-\Spop; -\DelMean, b) \der \Spop, \label{eq:polyProbaDel} \\
\proba [ \SpopNeu \given \Sphyclass] &= \proba [ -1 < \Spop < 1 \given \Sphyclass ] = \left( 1 - p_b \right) \int_{-1}^{0} f_{\Gamma}(-\Spop; -\DelMean, b) \der \Spop + p_b \int_{0}^{1} f_{e}(\Spop; \AdvMean) \der \Spop, \\
\proba [ \SpopBen \given \Sphyclass] &= \proba [ \Spop > 1 \given \Sphyclass] = p_b \int_{1}^{+\infty} f_{e}(\Spop; \AdvMean) \der \Spop. \label{eq:polyProbaAdv}
\end{align}

Rather than relying solely on currently segregating mutations to quantify selection, \textit{polyDFE} can leverage both divergence and polymorphism to estimate the parameters of the DFE.
We can thus add four more inputs to \textit{polyDFE}: $D \left( \Sphyclass \right) $, $L \left( \Sphyclass \right)$, $D_{\text{syn}}$ and $L_{\text{syn}}$ such as defined in the previous section.
Because the estimates of DFE are different with this method, we naturally obtained different values of $\proba [ \SpopDel \given  \Sphyclass] $, $\proba [ \SpopNeu \given \Sphyclass]$, and $\proba [ \SpopBen \given \Sphyclass]$.

\subsection{\textit{Precision} and \textit{recall}}
\label{subsec:precisison_recall}
For readability, we give here \textit{precision} and \textit{recall} for beneficial mutations ($\SphyBen$ and $\SpopBen$), but it can be obtained using the same derivation for the deleterious mutations ($\SphyDel$ and $\SpopDel$) and nearly-neutral mutations ($\SphyNeu$ and $\SpopNeu$).

\textit{Precision} is the proportion of mutations correctly predicted as beneficial ($\proba [ \SpopBen \cap  \SphyBen]$) out of all predicted as beneficial non-adaptive mutations ($\proba [ \SphyBen]$), which can be written as a conditional probability:

\begin{equation}
\frac{\proba [ \SpopBen  \cap  \SphyBen]}{\proba [ \SphyBen]} = \proba [ \SpopBen \given \SphyBen].
\label{eq:precision}
\end{equation}
Namely, \textit{precision} corresponds to the probability for a $\SphyBen$ mutation to be effectively beneficial at the population level ($\SpopBen$).
This probability, computed from eq.~\ref{eq:polyProbaAdv}, is obtained by restricting our analysis to SNPs that are predicted to be beneficial non-adaptive mutations (yellow fill for the category $\SphyBen$ in Fig~\ref{fig:homo-afr-results}D).

\textit{Recall} is the proportion of mutations correctly predicted as beneficial ($\proba [ \SpopBen \cap  \SphyBen]$) out of all beneficial mutations ($\proba [ \SpopBen]$), which can be written as a conditional probability:
\begin{equation}
\frac{\proba [ \SpopBen \cap  \SphyBen]}{\proba [ \SpopBen]} = \proba [ \SphyBen  \given \SpopBen ].
\end{equation}
Namely, \textit{recall} corresponds to the probability for a beneficial mutation at the population level ($\SpopBen$) to be a beneficial non-adaptive mutation ($\SphyBen$).
Using Bayes theorem, \textit{recall} can be re-written as:
\begin{equation}
\proba [\SphyBen \given \SpopBen] = \frac{\proba [\SpopBen \given \SphyBen] \times \proba[\SphyBen]}{\ProbaPopBen},
\label{eq:bayes}
\end{equation}

where $\proba [\SpopBen \given \SphyBen]$ and $\proba [ \SphyBen ]$ can be calculated using equations~\ref{eq:precision} and~\ref{eq:proba-dfe-mutsel}, respectively, and $\proba [ \SpopBen ]$ is the probability of a mutation to be beneficial at the level of the population, which can be computed from the law of total probabilities as:

\begin{equation}
\proba [ \SpopBen ] = \sum_{\Sphyclass \in \{\SphyDel, \SphyNeu, \SphyBen \} }\proba [\SpopBen \given \Sphyclass ] \times \proba [\Sphyclass ].
\label{eq:total_proba}
\end{equation}

\subsection{Correlation with effective population size (\texorpdfstring{$\bm{\Ne}$}{Nₑ})}
\label{subsec:correlation-diversity}
Genetic diversity estimator Watterson's $\thetaSyn$ was obtained for each population from the synonymous SFS as in Achaz~\cite{achaz_frequency_2009}.
For each popuation, $\Ne$ was estimated from the equation $\Ne=\thetaSyn / (4 \times u)$, where $u$ is the mutation rate per generation.
Estimates for $u$ were averaged per species across the pedigree-based estimation in~\cite{bergeron_evolution_2023} for \textit{Homo}, \textit{Bos}, \textit{Capra} and \textit{Chlorocebus}.
For \textit{Ovis} we used the estimated $u$ of \textit{Capra}.
For \textit{Equus}, we used $u$ as estimated in Orlando \textit{et al.}~\cite{orlando_recalibrating_2013} ($u=7.24\times10^{-9}$).
Because a correlation must account for phylogenetic relationship and non-independence of samples, we fitted a Phylogenetic Generalized Linear Model in \textit{R} with the method \texttt{pgls} with default settings from the package \texttt{caper}~\cite{orme_caper_2013}.
The mammalian dated tree was obtained from TimeTree~\cite{kumar_timetree_2017} and pruned to include only the species analysed in this study, with multi-furcation of the different populations from each species placed at the same divergence time as the species (section S4.1).

\section*{Data availability}
The data underlying this article are available at \url{https://doi.org/10.5281/zenodo.7878953}.
Snakemake pipeline, analysis scripts and documentation are available at \href{https://github.com/ThibaultLatrille/SelCoeff}{github.com/ThibaultLatrille/SelCoeff}.

\section*{Acknowledgements}
\label{sec:acknowledgment}
We gratefully acknowledge the help of Mélodie Bastian, Nicolas Lartillot, Carina Farah Mugal, Laurent Duret, Alexandre Reymond, Daniele Silvestro and Nicolas Gambardella for their advice and reviews concerning this manuscript.
This work was performed using the computing facilities of the CC LBBE/PRABI\@.
This study makes use of data generated by the NextGen Consortium.
Funding:
Université de Lausanne; Agence Nationale de la Recherche, Grant ANR-19-CE12-0019 / HotRec.


\section*{Competing interests}
The authors declare no conflicts of interest.

\section*{Author information}
Original idea: T.L.\ and J.J.;
Model conception: T.L., J.J.\ and N.S.;
Code: T.L.;
Data analyses: T.L.\ and J.J.;
Interpretation: T.L., J.J., D.A.H.\ and N.S.;
First draft: T.L.\ and J.J.;
Editing and revisions: T.L., J.J., D.A.H.\ and N.S.
Project management and funding: N.S\@.

\newpage
\section*{Supporting information}

\paragraph*{S1 File.}
\label{S1_File}
{\bf Supplementary appendix.} Contains 43 pages of supplementary information: model parameterization, supplementary figures (S1 Fig to S21 Fig) and supplementary tables (S1 Table to S15 Table).

\subsection*{List of Figures}
\paragraph*{S1 Fig.}
\label{S1_Fig}
{\bf Bayesian hierarchical model.}

\paragraph*{S2 Fig.}
\label{S2_Fig}
{\bf Examples of $\bm{\SphyBen}$ mutation sites in \textit{Chlorocebus sabaeus} (reversions).}

\paragraph*{S3 Fig.}
\label{S3_Fig}
{\bf Examples of $\bm{\SphyBen}$ mutation sites in \textit{Chlorocebus sabaeus}.}

\paragraph*{S4 Fig.}
\label{S4_Fig}
{\bf Enrichment in $\bm{\SphyBen}$ SNPs across all gene ontology terms.}

\paragraph*{S5 Fig.}
\label{S5_Fig}
{\bf Estimation of selection at the population scale for $\bm{\SphyDel}$ mutations.}

\paragraph*{S6 Fig.}
\label{S6_Fig}
{\bf Estimation of selection at the population scale for $\bm{\SphyNeu}$ mutations.}

\paragraph*{S7 Fig.}
\label{S7_Fig}
{\bf Estimation of selection at the population scale for $\bm{\SphyBen}$ mutations.}

\paragraph*{S8 Fig.}
\label{S8_Fig}
{\bf Distribution of scaled selection coefficients ($\bm{\Sphy}$).}

\paragraph*{S9 Fig.}
\label{S9_Fig}
{\bf Tree used for PGLM (Phylogenetic Generalized Linear Model).}

\paragraph*{S10 Fig.}
\label{S10_Fig}
{\bf Relationship between the proportion of deleterious mutations ($\bm{\SpopDel}$) and effective population size.}

\paragraph*{S11 Fig.}
\label{S11_Fig}
{\bf Relationship between the proportion of deleterious mutations ($\bm{\SpopNeu}$) and effective population size.}

\paragraph*{S12 Fig.}
\label{S12_Fig}
{\bf Relationship between the proportion of deleterious mutations ($\bm{\SpopBen}$) and effective population size.}

\paragraph*{S13 Fig.}
\label{S13_Fig}
{\bf $\bm{\proba[\SphyBen\given \SpopBen ]}$ as a function of distance to the sister species.}

\paragraph*{S14 Fig.}
\label{S14_Fig}
{\bf Ratio of $\bm{\proba[\Sphy]}$ over $\bm{\proba[-\Sphy]}$ for polymorphisms of the African population of \textit{Homo sapiens}.}

\paragraph*{S15 Fig.}
\label{S15_Fig}
{\bf Proportion of non-adaptive mutations among beneficial ones ($\bm{\proba{[}\SphyBen\given \SpopBen {]}}$) estimated on synthetic site-frequency spectra.}

\paragraph*{S16 Fig.}
\label{S16_Fig}
{\bf Site-frequency spectrum when masking CpG transitions.}

\paragraph*{S17 Fig.}
\label{S17_Fig}
{\bf Including substitutions in the terminal lineage to estimate $\bm{\Spop}$.}

\paragraph*{S18 Fig.}
\label{S18_Fig}
{\bf $\bm{\proba[\SphyBen\given \SpopBen ]}$ when including divergence to estimate $\bm{\Spop}$, as a function of distance to the sister species.}

\paragraph*{S19 Fig.}
\label{S19_Fig}
{\bf $\bm{\proba [ \SpopDel ]}$ when fitting a single or three DFEs.}

\paragraph*{S20 Fig.}
\label{S20_Fig}
{\bf $\bm{\proba [ \SpopNeu ]}$ when fitting a single or three DFEs.}

\paragraph*{S21 Fig.}
\label{S21_Fig}
{\bf $\bm{\proba [ \SpopBen ]}$ when fitting a single or three DFEs.}


\newpage
\subsection*{List of Tables}
\paragraph*{S1 Table.}
\label{S1_Table}
{\bf Probability of mutations and substitutions to be $\bm{\SphyDel}$, $\bm{\SphyNeu}$ or $\bm{\SphyBen}$.}

\paragraph*{S2 Table.}
\label{S2_Table}
{\bf $\bm{\dnds}$ for $\bm{\SphyDel}$, $\bm{\SphyNeu}$ or $\bm{\SphyBen}$.}

\paragraph*{S3 Table.}
\label{S3_Table}
{\bf $\bm{\dnds}$ over-estimation due to non-adaptive beneficial mutations.}

\paragraph*{S4 Table.}
\label{S4_Table}
{\bf Gene ontology enrichment in $\bm{\SphyBen}$ SNPs.}

\paragraph*{S5 Table.}
\label{S5_Table}
{\bf Terms associated with deleterious mutations $\bm{\SphyDel}$.}

\paragraph*{S6 Table.}
\label{S6_Table}
{\bf Terms associated with non-adaptive beneficial mutations $\bm{\SphyBen}$.}

\paragraph*{S7 Table.}
\label{S7_Table}
{\bf Probability for beneficial mutations to be non-adaptive ($\bm{\proba[\SphyBen\given \SpopBen ]}$).}

\paragraph*{S8 Table.}
\label{S8_Table}
{\bf $\bm{\proba[\SphyBen\given \SpopBen ]}$ when masking CpG transitions.}

\paragraph*{S9 Table.}
\label{S9_Table}
{\bf Precision and recall when masking CpG transitions.}

\paragraph*{S10 Table.}
\label{S10_Table}
{\bf $\bm{\proba[\SphyBen\given \SpopBen ]}$ excluding or not genes under adaptation.}

\paragraph*{S11 Table.}
\label{S11_Table}
{\bf $\bm{\proba[\SphyBen\given \SpopBen ]}$ when including divergence to estimate $\bm{\Spop}$.}

\paragraph*{S12 Table.}
\label{S12_Table}
{\bf Precision and recall when including divergence to estimate $\bm{\Spop}$.}

\paragraph*{S13 Table.}
\label{S13_Table}
{\bf $\bm{\proba[\SphyBen\given \SpopBen ]}$ excluding or not genes under adaptation, when including divergence to estimate $\bm{\Spop}$.}

\paragraph*{S14 Table.}
\label{S14_Table}
{\bf $\bm{\proba[\SphyBen\given \SpopBen ]}$ when using a discrete DFE to estimate $\bm{\Spop}$.}

\paragraph*{S15 Table.}
\label{S15_Table}
{\bf Precision and recall when using a discrete DFE to estimate $\bm{\Spop}$.}


% Either type in your references using
% \begin{thebibliography}{}
% \bibitem{}
% Text
% \end{thebibliography}
%
% or
%
% Compile your BiBTeX database using our plos2015.bst
% style file and paste the contents of your .bbl file
% here. See http://journals.plos.org/plosone/s/latex for
% step-by-step instructions.
%
\begin{thebibliography}{10}

\bibitem{darwin_origin_1859}
Darwin C.
\newblock On the {{Origin}} of {{Species}} by {{Means}} of {{Natural
Selection}}, or the {{Preservation}} of {{Favoured Races}} in the
{{Struggle}} for {{Life}}. vol. 220.
\newblock John Murray; 1859.

\bibitem{merrell_adaptive_1994}
Merrell DJ.
\newblock The {{Adaptive Seascape}}: {{The Mechanism}} of {{Evolution}}.
\newblock U of Minnesota Press; 1994.

\bibitem{gavrilets_adaptive_2009}
Gavrilets S, Losos JB.
\newblock Adaptive {{Radiation}}: {{Contrasting Theory}} with {{Data}}.
\newblock Science. 2009;323(5915):732--737.
\newblock doi:{10.1126/science.1157966}.

\bibitem{enard_viruses_2016}
Enard D, Cai L, Gwennap C, Petrov DA.
\newblock Viruses Are a Dominant Driver of Protein Adaptation in Mammals.
\newblock eLife. 2016;5:e12469.
\newblock doi:{10.7554/eLife.12469}.

\bibitem{mcdonald_adaptative_1991}
McDonald JH, Kreitman M.
\newblock Adaptative Protein Evolution at {{Adh}} Locus in {{Drosophila}}.
\newblock Nature. 1991;351(6328):652--654.
\newblock doi:{10.1038/351652a0}.

\bibitem{smith_adaptive_2002}
Smith NGC, {Eyre-Walker} A.
\newblock Adaptive Protein Evolution in {{Drosophila}}.
\newblock Nature. 2002;415(6875):1022--1024.
\newblock doi:{10.1038/4151022a}.

\bibitem{welch_estimating_2006}
Welch JJ.
\newblock Estimating the {{Genomewide Rate}} of {{Adaptive Protein Evolution}}
in {{Drosophila}}.
\newblock Genetics. 2006;173(2):821--837.
\newblock doi:{10.1534/genetics.106.056911}.

\bibitem{yang_statistical_2000}
Yang Z, Bielawski JR.
\newblock Statistical Methods for Detecting Molecular Adaptation.
\newblock Trends in Ecology and Evolution. 2000;15(12):496--503.
\newblock doi:{10.1016/S0169-5347(00)01994-7}.

\bibitem{eyre-walker_genomic_2006}
{Eyre-Walker} A.
\newblock The Genomic Rate of Adaptive Evolution.
\newblock Trends in Ecology \& Evolution. 2006;21(10):569--575.
\newblock doi:{10.1016/j.tree.2006.06.015}.

\bibitem{moutinho_variation_2019}
Moutinho AF, Bataillon T, Dutheil JY.
\newblock Variation of the Adaptive Substitution Rate between Species and
within Genomes.
\newblock Evolutionary Ecology. 2019;34(3):315--338.
\newblock doi:{10.1007/s10682-019-10026-z}.

\bibitem{lynch_mutation_2023}
Lynch M.
\newblock Mutation Pressure, Drift, and the Pace of Molecular Coevolution.
\newblock Proceedings of the National Academy of Sciences.
2023;120(27):e2306741120.
\newblock doi:{10.1073/pnas.2306741120}.

\bibitem{charlesworth_other_2007}
Charlesworth J, {Eyre-Walker} A.
\newblock The Other Side of the Nearly Neutral Theory, Evidence of Slightly
Advantageous Back-Mutations.
\newblock Proceedings of the National Academy of Sciences.
2007;104(43):16992--16997.
\newblock doi:{10.1073/pnas.0705456104}.

\bibitem{mustonen_fitness_2009}
Mustonen V, L{\"a}ssig M.
\newblock From Fitness Landscapes to Seascapes: Non-Equilibrium Dynamics of
Selection and Adaptation.
\newblock Trends in genetics. 2009;25(3):111--119.
\newblock doi:{10.1016/j.tig.2009.01.002}.

\bibitem{jones_shifting_2017}
Jones CT, Youssef N, Susko E, Bielawski JP.
\newblock Shifting Balance on a Static Mutation--Selection Landscape: A Novel
Scenario of Positive Selection.
\newblock Molecular Biology and Evolution. 2017;34(2):391--407.
\newblock doi:{10.1093/molbev/msw237}.

\bibitem{ohta_nearly_1992}
Ohta T.
\newblock The Nearly Neutral Theory of Molecular Evolution.
\newblock Annual Review of Ecology and Systematics. 1992;23(1992):263--286.

\bibitem{gillespie_ohta_1995}
Gillespie JH.
\newblock On {{Ohta}}'s Hypothesis: {{Most}} Amino Acid Substitutions Are
Deleterious.
\newblock Journal of Molecular Evolution. 1995;40(1):64--69.
\newblock doi:{10.1007/BF00166596}.

\bibitem{hartl_compensatory_1996}
Hartl DL, Taubes CH.
\newblock Compensatory Nearly Neutral Mutations: {{Selection}} without
Adaptation.
\newblock Journal of Theoretical Biology. 1996;182(3):303--309.
\newblock doi:{10.1006/jtbi.1996.0168}.

\bibitem{sella_application_2005}
Sella G, Hirsh AE.
\newblock The Application of Statistical Physics to Evolutionary Biology.
\newblock Proceedings of the National Academy of Sciences of the United States
of America. 2005;102(27):9541--9546.
\newblock doi:{10.1073/pnas.0501865102}.

\bibitem{cvijovic_fate_2015}
Cvijovi{\'c} I, Good BH, Jerison ER, Desai MM.
\newblock Fate of a Mutation in a Fluctuating Environment.
\newblock Proceedings of the National Academy of Sciences.
2015;112(36):E5021--E5028.
\newblock doi:{10.1073/pnas.1505406112}.

\bibitem{piganeau_estimating_2003}
Piganeau G, {Eyre-Walker} A.
\newblock Estimating the Distribution of Fitness Effects from {{DNA}} Sequence
Data: {{Implications}} for the Molecular Clock.
\newblock Proceedings of the National Academy of Sciences of the United States
of America. 2003;100(18):10335--10340.
\newblock doi:{10.1073/pnas.1833064100}.

\bibitem{chi_detecting_2020}
Chi PB, Kosater WM, Liberles DA.
\newblock Detecting {{Signatures}} of {{Positive Selection}} against a
{{Backdrop}} of {{Compensatory Processes}}.
\newblock Molecular Biology and Evolution. 2020;37(11):3353--3362.
\newblock doi:{10.1093/molbev/msaa161}.

\bibitem{bazykin_changing_2015}
Bazykin GA.
\newblock Changing Preferences: Deformation of Single Position Amino Acid
Fitness Landscapes and Evolution of Proteins.
\newblock Biology letters. 2015;11(10):20150315.
\newblock doi:{10.1098/rsbl.2015.0315}.

\bibitem{moses_inferring_2009}
Moses AM, Durbin R.
\newblock Inferring {{Selection}} on {{Amino Acid Preference}} in {{Protein
Domains}}.
\newblock Molecular Biology and Evolution. 2009;26(3):527--536.
\newblock doi:{10.1093/molbev/msn286}.

\bibitem{fischer_germline_2011}
Fischer A, Greenman C, Mustonen V.
\newblock Germline {{Fitness-Based Scoring}} of {{Cancer Mutations}}.
\newblock Genetics. 2011;188(2):383--393.
\newblock doi:{10.1534/genetics.111.127480}.

\bibitem{chen_hunting_2021}
Chen J, Bataillon T, Gl{\'e}min S, Lascoux M.
\newblock Hunting for Beneficial Mutations: Conditioning on {{SIFT}} Scores
When Estimating the Distribution of Fitness Effect of New Mutations.
\newblock Genome Biology and Evolution. 2021;doi:{10.1093/gbe/evab151}.

\bibitem{halpern_evolutionary_1998}
Halpern AL, Bruno WJ.
\newblock Evolutionary Distances for Protein-Coding Sequences: Modeling
Site-Specific Residue Frequencies.
\newblock Molecular Biology and Evolution. 1998;15(7):910--917.
\newblock doi:{10.1093/oxfordjournals.molbev.a025995}.

\bibitem{mccandlish_modeling_2014}
McCandlish DM, Stoltzfus A.
\newblock Modeling Evolution Using the Probability of Fixation: {{History}} and
Implications.
\newblock Quarterly Review of Biology. 2014;89(3):225--252.
\newblock doi:{10.1086/677571}.

\bibitem{rodrigue_mechanistic_2010}
Rodrigue N, Philippe H.
\newblock Mechanistic Revisions of Phenomenological Modeling Strategies in
Molecular Evolution.
\newblock Trends in Genetics. 2010;26(6):248--252.
\newblock doi:{10.1016/j.tig.2010.04.001}.

\bibitem{tamuri_estimating_2012}
Tamuri AU, Goldstein RA.
\newblock Estimating the Distribution of Selection Coefficients from
Phylogenetic Data Using Sitewise Mutation-Selection Models.
\newblock Genetics. 2012;190(3):1101--1115.
\newblock doi:{10.1534/genetics.111.136432}.

\bibitem{rodrigue_detecting_2017}
Rodrigue N, Lartillot N.
\newblock Detecting Adaptation in Protein-Coding Genes Using a {{Bayesian}}
Site- Heterogeneous Mutation-Selection Codon Substitution Model.
\newblock Molecular Biology and Evolution. 2017;34(1):204--214.
\newblock doi:{10.1093/molbev/msw220}.

\bibitem{foley_genomic_2023}
Foley NM, Mason VC, Harris AJ, Bredemeyer KR, Damas J, Lewin HA, et~al.
\newblock A Genomic Timescale for Placental Mammal Evolution.
\newblock Science. 2023;380(6643):eabl8189.
\newblock doi:{10.1126/science.abl8189}.

\bibitem{howe_ensembl_2021}
Howe KL, Achuthan P, Allen J, Allen J, {Alvarez-Jarreta} J, Amode MR, et~al.
\newblock Ensembl 2021.
\newblock Nucleic Acids Research. 2021;49(D1):D884--D891.
\newblock doi:{10.1093/nar/gkaa942}.

\bibitem{ranwez_orthomam_2007}
Ranwez V, Delsuc F, Ranwez S, Belkhir K, Tilak MK, Douzery EJ.
\newblock {{OrthoMaM}}: {{A}} Database of Orthologous Genomic Markers for
Placental Mammal Phylogenetics.
\newblock BMC Evolutionary Biology. 2007;7(1):1--12.
\newblock doi:{10.1186/1471-2148-7-241}.

\bibitem{scornavacca_orthomam_2019}
Scornavacca C, Belkhir K, Lopez J, Dernat R, Delsuc F, Douzery EJP, et~al.
\newblock {{OrthoMaM}} V10: {{Scaling-up}} Orthologous Coding Sequence and Exon
Alignments with More than One Hundred Mammalian Genomes.
\newblock Molecular Biology and Evolution. 2019;36(4):861--862.
\newblock doi:{10.1093/molbev/msz015}.

\bibitem{rodrigue_mutationselection_2010}
Rodrigue N, Philippe H, Lartillot N.
\newblock Mutation-Selection Models of Coding Sequence Evolution with
Site-Heterogeneous Amino Acid Fitness Profiles.
\newblock Proceedings of the National Academy of Sciences of the United States
of America. 2010;107(10):4629--34.
\newblock doi:{10.1073/pnas.0910915107}.

\bibitem{landrum_clinvar_2018}
Landrum MJ, Lee JM, Benson M, Brown GR, Chao C, Chitipiralla S, et~al.
\newblock {{ClinVar}}: Improving Access to Variant Interpretations and
Supporting Evidence.
\newblock Nucleic Acids Research. 2018;46(D1):D1062--D1067.
\newblock doi:{10.1093/nar/gkx1153}.

\bibitem{grimm_evaluation_2015}
Grimm DG, Azencott CA, Aicheler F, Gieraths U, MacArthur DG, Samocha KE, et~al.
\newblock The {{Evaluation}} of {{Tools Used}} to {{Predict}} the {{Impact}} of
{{Missense Variants Is Hindered}} by {{Two Types}} of {{Circularity}}.
\newblock Human Mutation. 2015;36(5):513--523.
\newblock doi:{10.1002/humu.22768}.

\bibitem{sullivan_leveraging_2023}
Sullivan PF, Meadows JRS, Gazal S, Phan BN, Li X, Genereux DP, et~al.
\newblock Leveraging Base-Pair Mammalian Constraint to Understand Genetic
Variation and Human Disease.
\newblock Science. 2023;380(6643):eabn2937.
\newblock doi:{10.1126/science.abn2937}.

\bibitem{eyre-walker_distribution_2006}
{Eyre-Walker} A, Woolfit M, Phelps T.
\newblock The Distribution of Fitness Effects of New Deleterious Amino Acid
Mutations in Humans.
\newblock Genetics. 2006;173(2):891--900.
\newblock doi:{10.1534/genetics.106.057570}.

\bibitem{eyre-walker_estimating_2009}
{Eyre-Walker} A, Keightley PD.
\newblock Estimating the {{Rate}} of {{Adaptive Molecular Evolution}} in the
{{Presence}} of {{Slightly Deleterious Mutations}} and {{Population Size
Change}}.
\newblock Molecular Biology and Evolution. 2009;26(9):2097--2108.
\newblock doi:{10.1093/molbev/msp119}.

\bibitem{galtier_adaptive_2016}
Galtier N.
\newblock Adaptive Protein Evolution in Animals and the Effective Population
Size Hypothesis.
\newblock PLoS Genetics. 2016;12(1):e1005774.
\newblock doi:{10.1371/journal.pgen.1005774}.

\bibitem{tataru_inference_2017}
Tataru P, Mollion M, Gl{\'e}min S, Bataillon T.
\newblock Inference of Distribution of Fitness Effects and Proportion of
Adaptive Substitutions from Polymorphism Data.
\newblock Genetics. 2017;207(3):1103--1119.
\newblock doi:{10.1534/genetics.117.300323}.

\bibitem{bergeron_evolution_2023}
Bergeron LA, Besenbacher S, Zheng J, Li P, Bertelsen MF, Quintard B, et~al.
\newblock Evolution of the Germline Mutation Rate across Vertebrates.
\newblock Nature. 2023; p. 1--7.
\newblock doi:{10.1038/s41586-023-05752-y}.

\bibitem{orlando_recalibrating_2013}
Orlando L, Ginolhac A, Zhang G, Froese D, Albrechtsen A, Stiller M, et~al.
\newblock Recalibrating {{Equus}} Evolution Using the Genome Sequence of an
Early {{Middle Pleistocene}} Horse.
\newblock Nature. 2013;499(7456):74--78.
\newblock doi:{10.1038/nature12323}.

\bibitem{achaz_frequency_2009}
Achaz G.
\newblock Frequency {{Spectrum Neutrality Tests}}: {{One}} for {{All}} and
{{All}} for {{One}}.
\newblock Genetics. 2009;183(1):249--258.
\newblock doi:{10.1534/genetics.109.104042}.

\bibitem{latrille_genes_2023}
Latrille T, Rodrigue N, Lartillot N.
\newblock Genes and Sites under Adaptation at the Phylogenetic Scale Also
Exhibit Adaptation at the Population-Genetic Scale.
\newblock Proceedings of the National Academy of Sciences of the United States
of America. 2023;120(11):e2214977120.
\newblock doi:{10.1073/pnas.2214977120}.

\bibitem{ng_sift_2003}
Ng PC, Henikoff S.
\newblock {{SIFT}}: Predicting Amino Acid Changes That Affect Protein Function.
\newblock Nucleic Acids Research. 2003;31(13):3812--3814.
\newblock doi:{10.1093/nar/gkg509}.

\bibitem{vaser_sift_2016}
Vaser R, Adusumalli S, Leng SN, Sikic M, Ng PC.
\newblock {{SIFT}} Missense Predictions for Genomes.
\newblock Nature Protocols. 2016;11(1):1--9.
\newblock doi:{10.1038/nprot.2015.123}.

\bibitem{keightley_what_2010}
Keightley PD, {Eyre-Walker} A.
\newblock What Can We Learn about the Distribution of Fitness Effects of New
Mutations from {{DNA}} Sequence Data?
\newblock Philosophical Transactions of the Royal Society B: Biological
Sciences. 2010;365(1544):1187--1193.
\newblock doi:{10.1098/rstb.2009.0266}.

\bibitem{rice_evolutionarily_2015}
Rice DP, Good BH, Desai MM.
\newblock The {{Evolutionarily Stable Distribution}} of {{Fitness Effects}}.
\newblock Genetics. 2015;200(1):321--329.
\newblock doi:{10.1534/genetics.114.173815}.

\bibitem{mustonen_fitness_2010}
Mustonen V, L{\"a}ssig M.
\newblock Fitness Flux and Ubiquity of Adaptive Evolution.
\newblock Proceedings of the National Academy of Sciences.
2010;107(9):4248--4253.
\newblock doi:{10.1073/pnas.0907953107}.

\bibitem{eyre-walker_how_2014}
{Eyre-Walker} A, {Eyre-Walker} YC.
\newblock How Much of the Variation in the Mutation Rate along the Human Genome
Can Be Explained?
\newblock G3: Genes, Genomes, Genetics. 2014;4(9):1667--1670.
\newblock doi:{10.1534/g3.114.012849}.

\bibitem{mustonen_molecular_2008}
Mustonen V, L{\"a}ssig M.
\newblock Molecular {{Evolution}} under {{Fitness Fluctuations}}.
\newblock Physical Review Letters. 2008;100(10):108101.
\newblock doi:{10.1103/PhysRevLett.100.108101}.

\bibitem{latrille_inferring_2021}
Latrille T, Lanore V, Lartillot N.
\newblock Inferring Long-Term Effective Population Size with
Mutation--Selection Models.
\newblock Molecular Biology and Evolution. 2021;38(10):4573--4587.
\newblock doi:{10.1093/molbev/msab160}.

\bibitem{huber_determining_2017}
Huber CD, Kim BY, Marsden CD, Lohmueller KE.
\newblock Determining the Factors Driving Selective Effects of New
Nonsynonymous Mutations.
\newblock Proceedings of the National Academy of Sciences.
2017;114(17):4465--4470.
\newblock doi:{10.1073/pnas.1619508114}.

\bibitem{lanfear_population_2014}
Lanfear R, Kokko H, {Eyre-Walker} A.
\newblock Population Size and the Rate of Evolution.
\newblock Trends in Ecology and Evolution. 2014;29(1):33--41.
\newblock doi:{10.1016/j.tree.2013.09.009}.

\bibitem{platt_protein_2018}
Platt A, Weber CC, Liberles DA.
\newblock Protein Evolution Depends on Multiple Distinct Population Size
Parameters.
\newblock BMC Evolutionary Biology. 2018;18(1):1--9.
\newblock doi:{10.1186/s12862-017-1085-x}.

\bibitem{starr_epistasis_2016}
Starr TN, Thornton JW.
\newblock Epistasis in Protein Evolution.
\newblock Protein Science. 2016;25(7):1204--1218.
\newblock doi:{10.1002/pro.2897}.

\bibitem{morcos_directcoupling_2011}
Morcos F, Pagnani A, Lunt B, Bertolino A, Marks DS, Sander C, et~al.
\newblock Direct-Coupling Analysis of Residue Coevolution Captures Native
Contacts across Many Protein Families.
\newblock Proceedings of the National Academy of Sciences.
2011;108(49):E1293--E1301.
\newblock doi:{10.1073/pnas.1111471108}.

\bibitem{marks_protein_2012}
Marks DS, Hopf TA, Sander C.
\newblock Protein Structure Prediction from Sequence Variation.
\newblock Nature Biotechnology. 2012;30(11):1072--1080.
\newblock doi:{10.1038/nbt.2419}.

\bibitem{goldstein_nonadaptive_2015}
Goldstein RA, Pollard ST, Shah SD, Pollock DD.
\newblock Nonadaptive {{Amino Acid Convergence Rates Decrease}} over {{Time}}.
\newblock Molecular Biology and Evolution. 2015;32(6):1373--1381.
\newblock doi:{10.1093/molbev/msv041}.

\bibitem{goldstein_sequence_2017}
Goldstein RA, Pollock DD.
\newblock Sequence Entropy of Folding and the Absolute Rate of Amino Acid
Substitutions.
\newblock Nature Ecology \& Evolution. 2017;1(12):1923--1930.
\newblock doi:{10.1038/s41559-017-0338-9}.

\bibitem{park_epistatic_2022}
Park Y, Metzger BPH, Thornton JW.
\newblock Epistatic Drift Causes Gradual Decay of Predictability in Protein
Evolution.
\newblock Science. 2022;376(6595):823--830.
\newblock doi:{10.1126/science.abn6895}.

\bibitem{ashenberg_mutational_2013}
Ashenberg O, Gong LI, Bloom JD.
\newblock Mutational Effects on Stability Are Largely Conserved during Protein
Evolution.
\newblock Proceedings of the National Academy of Sciences of the United States
of America. 2013;110(52):21071--21076.
\newblock doi:{10.1073/pnas.1314781111}.

\bibitem{doud_sitespecific_2015}
Doud MB, Ashenberg O, Bloom JD.
\newblock Site-{{Specific Amino Acid Preferences Are Mostly Conserved}} in
{{Two Closely Related Protein Homologs}}.
\newblock Molecular Biology and Evolution. 2015;32(11):2944--2960.
\newblock doi:{10.1093/molbev/msv167}.

\bibitem{bloom_identification_2017}
Bloom JD.
\newblock Identification of Positive Selection in Genes Is Greatly Improved by
Using Experimentally Informed Site-Specific Models.
\newblock Biology Direct. 2017;12(1):1--24.
\newblock doi:{10.1186/s13062-016-0172-z}.

\bibitem{youssef_consequences_2020}
Youssef N, Susko E, Bielawski JP.
\newblock Consequences of Stability-Induced Epistasis for Substitution Rates.
\newblock Molecular Biology and Evolution. 2020;doi:{10.1093/molbev/msaa151}.

\bibitem{vigue_deciphering_2022}
Vigu{\'e} L, Croce G, Petitjean M, Rupp{\'e} E, Tenaillon O, Weigt M.
\newblock Deciphering Polymorphism in 61,157 {{Escherichia}} Coli Genomes via
Epistatic Sequence Landscapes.
\newblock Nature Communications. 2022;13(1):4030.
\newblock doi:{10.1038/s41467-022-31643-3}.

\bibitem{kimura_evolutionary_1968}
Kimura M.
\newblock Evolutionary Rate at the Molecular Level.
\newblock Nature. 1968;217(5129):624--626.

\bibitem{gillespie_substitution_1994}
Gillespie JH.
\newblock Substitution Processes in Molecular Evolution. {{III}}.
{{Deleterious}} Alleles.
\newblock Genetics. 1994;138(3):943--952.
\newblock doi:{10.1093/genetics/138.3.943}.

\bibitem{jensen_importance_2019}
Jensen JD, Payseur BA, Stephan W, Aquadro CF, Lynch M, Charlesworth D, et~al.
\newblock The Importance of the {{Neutral Theory}} in 1968 and 50 Years on:
{{A}} Response to {{Kern}} and {{Hahn}} 2018.
\newblock Evolution. 2019;73(1):111--114.
\newblock doi:{10.1111/evo.13650}.

\bibitem{galtier_half_2024}
Galtier N.
\newblock Half a {{Century}} of {{Controversy}}: {{The
Neutralist}}/{{Selectionist Debate}} in {{Molecular Evolution}}.
\newblock Genome Biology and Evolution. 2024;16(2):evae003.
\newblock doi:{10.1093/gbe/evae003}.

\bibitem{galtier_adaptation_2007}
Galtier N, Duret L.
\newblock Adaptation or Biased Gene Conversion? {{Extending}} the Null
Hypothesis of Molecular Evolution.
\newblock Trends in Genetics. 2007;23(6):273--277.
\newblock doi:{10.1016/j.tig.2007.03.011}.

\bibitem{rousselle_influence_2019}
Rousselle M, Laverr{\'e} A, Figuet E, Nabholz B, Galtier N.
\newblock Influence of {{Recombination}} and {{GC-biased Gene Conversion}} on
the {{Adaptive}} and {{Nonadaptive Substitution Rate}} in {{Mammals}} versus
{{Birds}}.
\newblock Molecular Biology and Evolution. 2019;36(3):458--471.
\newblock doi:{10.1093/molbev/msy243}.

\bibitem{joseph_increased_2024a}
Joseph J.
\newblock Increased {{Positive Selection}} in {{Highly Recombining Genes Does}}
Not {{Necessarily Reflect}} an {{Evolutionary Advantage}} of
{{Recombination}}.
\newblock Molecular Biology and Evolution. 2024; p. msae107.
\newblock doi:{10.1093/molbev/msae107}.

\bibitem{ohta_development_1996}
Ohta T, Gillespie JH.
\newblock Development of {{Neutral}} and {{Nearly Neutral Theories}}.
\newblock Theoretical Population Biology. 1996;49(2):128--142.
\newblock doi:{10.1006/tpbi.1996.0007}.

\bibitem{spielman_relationship_2015}
Spielman SJ, Wilke CO.
\newblock The Relationship between {{dN}}/{{dS}} and Scaled Selection
Coefficients.
\newblock Molecular biology and evolution. 2015;32(4):1097--1108.
\newblock doi:{10.1093/molbev/msv003}.

\bibitem{dosreis_how_2015}
Dos~Reis M.
\newblock How to Calculate the Non-Synonymous to Synonymous Rate Ratio of
Protein-Coding Genes under the Fisher-Wright Mutation-Selection Framework.
\newblock Biology Letters. 2015;11(4):20141031.
\newblock doi:{10.1098/rsbl.2014.1031}.

\bibitem{rodrigue_bayesian_2021}
Rodrigue N, Latrille T, Lartillot N.
\newblock A {{Bayesian}} Mutation-Selection Framework for Detecting
Site-Specific Adaptive Evolution in Protein-Coding Genes.
\newblock Molecular Biology and Evolution. 2021;38(3):1199--1208.
\newblock doi:{10.1093/molbev/msaa265}.

\bibitem{tamuri_mutationselection_2021}
Tamuri AU, {dos Reis} M.
\newblock A Mutation-Selection Model of Protein Evolution under Persistent
Positive Selection.
\newblock Molecular Biology and Evolution. 2021;doi:{10.1093/molbev/msab309}.

\bibitem{kazmi_detecting_2019}
Kazmi SO, Rodrigue N.
\newblock Detecting Amino Acid Preference Shifts with Codon-Level
Mutation-Selection Mixture Models.
\newblock BMC Evolutionary Biology. 2019;19(1):62.
\newblock doi:{10.1186/s12862-019-1358-7}.

\bibitem{stolyarova_senescence_2020}
Stolyarova AV, Nabieva E, Ptushenko VV, Favorov AV, Popova AV, Neverov AD,
et~al.
\newblock Senescence and Entrenchment in Evolution of Amino Acid Sites.
\newblock Nature Communications. 2020;11(1):4603.
\newblock doi:{10.1038/s41467-020-18366-z}.

\bibitem{douzery_orthomam_2014}
Douzery EJP, Scornavacca C, Romiguier J, Belkhir K, Galtier N, Delsuc F, et~al.
\newblock {{OrthoMaM}} v8: {{A}} Database of Orthologous Exons and Coding
Sequences for Comparative Genomics in Mammals.
\newblock Molecular Biology and Evolution. 2014;31(7):1923--1928.
\newblock doi:{10.1093/molbev/msu132}.

\bibitem{yang_mutationselection_2008}
Yang Z, Nielsen R.
\newblock Mutation-Selection Models of Codon Substitution and Their Use to
Estimate Selective Strengths on Codon Usage.
\newblock Molecular Biology and Evolution. 2008;25(3):568--579.
\newblock doi:{10.1093/molbev/msm284}.

\bibitem{wright_evolution_1931}
Wright S.
\newblock Evolution in {{Mendelian}} Populations.
\newblock Genetics. 1931;16(2):97--159.
\newblock doi:{10.1093/genetics/16.2.97}.

\bibitem{fisher_genetical_1930}
Fisher RA.
\newblock The {{Genetical Theory}} of {{Natural Selection}}.
\newblock The Clarendon Press; 1930.

\bibitem{rodrigue_bayesian_2008}
Rodrigue N, Lartillot N, Philippe H.
\newblock Bayesian Comparisons of Codon Substitution Models.
\newblock Genetics. 2008;180(3):1579--1591.
\newblock doi:{10.1534/genetics.108.092254}.

\bibitem{lartillot_bayesian_2004}
Lartillot N.
\newblock A Bayesian Mixture Model for Across-Site Heterogeneities in the
Amino-Acid Replacement Process.
\newblock Molecular biology and evolution. 2004;21(6):1095--1109.
\newblock doi:{10.1093/molbev/msh112}.

\bibitem{lartillot_inference_2013}
Lartillot N.
\newblock Inf{\'e}rence Probabiliste Pour La Phylog{\'e}nie, La G{\'e}nomique
Comparative et Les Sciences de La Macro-{\'E}volution; 2013.

\bibitem{alabri_whole_2020}
Al~Abri MA, Holl HM, Kalla SE, Sutter NB, Brooks SA.
\newblock Whole Genome Detection of Sequence and Structural Polymorphism in Six
Diverse Horses.
\newblock PLoS ONE. 2020;15(4):e0230899.
\newblock doi:{10.1371/journal.pone.0230899}.

\bibitem{sherry_dbsnp_2001}
Sherry ST, Ward MH, Kholodov M, Baker J, Phan L, Smigielski EM, et~al.
\newblock {{dbSNP}}: The {{NCBI}} Database of Genetic Variation.
\newblock Nucleic Acids Research. 2001;29(1):308--311.
\newblock doi:{10.1093/nar/29.1.308}.

\bibitem{svardal_ancient_2017}
Svardal H, Jasinska AJ, Apetrei C, Coppola G, Huang Y, Schmitt CA, et~al.
\newblock Ancient Hybridization and Strong Adaptation to Viruses across
{{African}} Vervet Monkey Populations.
\newblock Nature Genetics. 2017;49(12):1705--1713.
\newblock doi:{10.1038/ng.3980}.

\bibitem{zheng-bradley_alignment_2017}
{Zheng-Bradley} X, Streeter I, Fairley S, Richardson D, Clarke L, Flicek P,
et~al.
\newblock Alignment of 1000 {{Genomes Project}} Reads to Reference Assembly
{{GRCh38}}.
\newblock GigaScience. 2017;6(7):gix038.
\newblock doi:{10.1093/gigascience/gix038}.

\bibitem{keightley_inferring_2018}
Keightley PD, Jackson BC.
\newblock Inferring the Probability of the Derived vs the Ancestral Allelic
State at a Polymorphic Site.
\newblock Genetics. 2018;209(3):897--906.
\newblock doi:{10.1534/genetics.118.301120}.

\bibitem{ashkenazy_fastml_2012}
Ashkenazy H, Penn O, {Doron-Faigenboim} A, Cohen O, Cannarozzi G, Zomer O,
et~al.
\newblock {{FastML}}: A Web Server for Probabilistic Reconstruction of
Ancestral Sequences.
\newblock Nucleic Acids Research. 2012;40(W1):W580--W584.
\newblock doi:{10.1093/nar/gks498}.

\bibitem{tataru_polydfe_2020}
Tataru P, Bataillon T.
\newblock {{polyDFE}}: {{Inferring}} the Distribution of Fitness Effects and
Properties of Beneficial Mutations from Polymorphism Data.
\newblock In: Methods in {{Molecular Biology}}. vol. 2090. Humana Press Inc.;
2020. p. 125--146.

\bibitem{orme_caper_2013}
Orme D, Freckleton R, Thomas G, Petzoldt T, Fritz S, Isaac N, et~al.
\newblock The Caper Package: Comparative Analysis of Phylogenetics and
Evolution in {{R}}.
\newblock R package version. 2013;5(2):1--36.

\bibitem{kumar_timetree_2017}
Kumar S, Stecher G, Suleski M, Hedges SB.
\newblock {{TimeTree}}: A Resource for Timelines, Timetrees, and Divergence
Times.
\newblock Molecular Biology and Evolution. 2017;34(7):1812--1819.
\newblock doi:{10.1093/molbev/msx116}.

\end{thebibliography}

\end{document}