%TC:ignore
\documentclass{article}

\usepackage{xcolor, colortbl}
\definecolor{BLUELINK}{HTML}{0645AD}
\definecolor{DARKBLUELINK}{HTML}{0B0080}
\PassOptionsToPackage{hyphens}{url}
\usepackage[colorlinks=false]{hyperref}
% for linking between references, figures, TOC, etc in the pdf document
\hypersetup{colorlinks,
    linkcolor=DARKBLUELINK,
    anchorcolor=DARKBLUELINK,
    citecolor=DARKBLUELINK,
    filecolor=DARKBLUELINK,
    menucolor=DARKBLUELINK,
    urlcolor=BLUELINK
} % Color citation links in purple
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{naturalnames}{hyperref}

\usepackage[margin=60pt]{geometry}
\usepackage{amssymb,amsfonts,amsmath,amsthm,mathtools}
\usepackage{lmodern}
\usepackage{bm,bbold}
\usepackage{verbatim}
\usepackage{float}
\usepackage{listings, enumerate, enumitem}
\usepackage[export]{adjustbox}
\usepackage{tabu}
\usepackage{longtable}
\tabulinesep=0.6mm
\newcommand\cellwidth{\TX@col@width}
\usepackage{hhline}
\setlength{\arrayrulewidth}{1.2pt}
\usepackage{multicol,multirow,array}
\usepackage{etoolbox}
\AtBeginEnvironment{tabu}{\footnotesize}
\usepackage{booktabs}
\usepackage{graphicx}
\pdfinclusioncopyfonts=1

\newcommand{\UniDimArray}[1]{\bm{#1}}
\newcommand{\BiDimArray}[1]{\bm{#1}}
\DeclareMathOperator{\E}{\mathbb{E}}
\DeclareMathOperator{\Var}{\mathrm{Var}}
\newcommand{\der}{\mathrm{d}}
\newcommand{\e}{\mathrm{e}}
\newcommand{\avg}[1]{\left< #1 \right>} % for average
\newcommand{\Ne}{N_{\mathrm{e}}}
\newcommand{\proba}{\mathbb{P}}
\newcommand{\pfix}{\proba_{\mathrm{fix}}}

\newcommand{\Sphy}{S}
\newcommand{\SphyMean}{\overline{\Sphy}}
\newcommand{\divStrongDel}{\Sphy < -3}
\newcommand{\divDel}{-3 < \Sphy < -1}
\newcommand{\divWeakDel}{-1 < \Sphy < 0}
\newcommand{\divWeakAdv}{0 < \Sphy < 1}
\newcommand{\divAdv}{ \Sphy > 1}
\newcommand{\PdivStrongDel}{\proba \left[ \divStrongDel \right]}
\newcommand{\PdivDel}{\proba \left[ \divDel \right]}
\newcommand{\PdivWeakDel}{\proba \left[ \divWeakDel \right]}
\newcommand{\PdivWeakAdv}{\proba \left[ \divWeakAdv \right]}
\newcommand{\PdivAdv}{\proba \left[ \divAdv \right]}

\newcommand{\Spop}{\beta}
\newcommand{\SpopMean}{\overline{\Spop}}
\newcommand{\polyDel}{\Spop < -1}
\newcommand{\polyNeutral}{-1 < \Spop < 1}
\newcommand{\polyAdv}{ \Spop > 1}
\newcommand{\PpolyDel}{\proba \left[ \polyDel \right]}
\newcommand{\PpolyNeutral}{\proba \left[ \polyNeutral \right]}
\newcommand{\PpolyAdv}{\proba \left[ \polyAdv \right]}

\title{Selection coefficients at the phylogenetic and population-genetic scale}

\author{
    \large
    T. {Latrille}$^{1}$, J. {Joseph}$^{2}$, N. {Salamin}$^{1}$\\
    $^{1}$Université de Lausanne, Lausanne, Switzerland
    $^{1}$Université de Lyon, Lyon, France
}

\begin{document}
    \maketitle
    \tableofcontents
    \clearpage


    \section{Site-specific mutation-selection codon models}

    In BayesCode (\url{https://github.com/ThibaultLatrille/bayescode}), mutation-selection codon models are obtained by running \textit{mutselomega} for 2000 points of MCMC with the options:
    \begin{scriptsize}
        \begin{verbatim}
        mutselomega ---omegashift 0.0 --ncat 30 -a my_alignment.phy -t my_tree.newick -u 2000 my_genename
        \end{verbatim}
    \end{scriptsize}
    The collection of site-specific fitness profiles ($\UniDimArray{F^{(i)}}, \forall i$) are then obtained by running \textit{readmutselomega}, reading 1000 points of MCMC (first 1000 are considered as burn-in) with the options:
    \begin{scriptsize}
        \begin{verbatim}
        readmutselomega --every 1 --until 2000 --burnin 1000 --ss my_genename
        \end{verbatim}
    \end{scriptsize}
    The gene-specific mutation matrix ($\UniDimArray{\mu}$) is also obtained by running \textit{readmutselomega}, reading 1000 points of MCMC (first 1000 are considered as burn-in) with the options:
    \begin{scriptsize}
        \begin{verbatim}
        readmutselomega --every 1 --until 2000 --burnin 1000 --nuc my_genename
        \end{verbatim}
    \end{scriptsize}

    \subsection{Summary table}

    \begin{center}
        \begin{longtable}{|l|l|r|r|r|r|r|}
            \toprule
            Population           & Species             & Tajima $\theta_{\pi}$ & $\mathbb{P}_{div}(S > 0)$ & $\mathbb{P}_{obs}[S>0]$ & $\mathbb{P} [ \beta > 0 ]$ & $\mathbb{P} [ \beta > 0  | S>0]$ \\
            \midrule
            \endhead
            \midrule
            \multicolumn{7}{r}{{Continued on next page}} \\
            \midrule
            \endfoot

            \bottomrule
            \endlastfoot
            Equus c.             & Equus caballus      & $ 0.002$              & $ 0.301$                  & $ 0.243$                & $ 0.113$                   & $ 0.793$                       \\
            Canis f.             & Canis familiaris    & $ 0.003$              & $ 0.421$                  & $ 0.181$                & $ 0.048$                   & $ 0.761$                       \\
            Iran                 & Bos taurus          & $ 0.005$              & $ 0.249$                  & $ 0.264$                & $ 0.098$                   & $ 0.605$                       \\
            Uganda               & Bos taurus          & $ 0.006$              & $ 0.254$                  & $ 0.249$                & $ 0.095$                   & $ 0.630$                       \\
            Australia            & Capra hircus        & $ 0.003$              & $ 0.286$                  & $ 0.265$                & $ 0.096$                   & $ 0.512$                       \\
            France               & Capra hircus        & $ 0.004$              & $ 0.285$                  & $ 0.274$                & $ 0.092$                   & $ 0.529$                       \\
            Iran (C. aegagrus)   & Capra hircus        & $ 0.004$              & $ 0.284$                  & $ 0.270$                & $ 0.095$                   & $ 0.455$                       \\
            Iran                 & Capra hircus        & $ 0.004$              & $ 0.297$                  & $ 0.239$                & $ 0.080$                   & $ 0.500$                       \\
            Italy                & Capra hircus        & $ 0.004$              & $ 0.285$                  & $ 0.265$                & $ 0.087$                   & $ 0.500$                       \\
            Morocco              & Capra hircus        & $ 0.004$              & $ 0.296$                  & $ 0.200$                & $ 0.080$                   & $ 0.509$                       \\
            Iran                 & Ovis aries          & $ 0.008$              & $ 0.229$                  & $ 0.259$                & $ 0.060$                   & $ 0.366$                       \\
            Iran (O. orientalis) & Ovis aries          & $ 0.009$              & $ 0.226$                  & $ 0.272$                & $ 0.067$                   & $ 0.344$                       \\
            Iran (O. vignei)     & Ovis aries          & $ 0.008$              & $ 0.235$                  & $ 0.307$                & $ 0.075$                   & $ 0.464$                       \\
            Various              & Ovis aries          & $ 0.009$              & $ 0.226$                  & $ 0.279$                & $ 0.065$                   & $ 0.378$                       \\
            Morocco              & Ovis aries          & $ 0.008$              & $ 0.225$                  & $ 0.212$                & $ 0.054$                   & $ 0.377$                       \\
            Barbados             & Chlorocebus sabaeus & $ 0.004$              & $ 0.314$                  & $ 0.258$                & $ 0.109$                   & $ 0.670$                       \\
            Central Afr. Rep.    & Chlorocebus sabaeus & $ 0.006$              & $ 0.315$                  & $ 0.264$                & $ 0.094$                   & $ 0.619$                       \\
            Ethiopia             & Chlorocebus sabaeus & $ 0.005$              & $ 0.307$                  & $ 0.252$                & $ 0.107$                   & $ 0.660$                       \\
            Gambia               & Chlorocebus sabaeus & $ 0.005$              & $ 0.303$                  & $ 0.249$                & $ 0.085$                   & $ 0.666$                       \\
            Kenya                & Chlorocebus sabaeus & $ 0.006$              & $ 0.311$                  & $ 0.272$                & $ 0.095$                   & $ 0.602$                       \\
            Nevis                & Chlorocebus sabaeus & $ 0.004$              & $ 0.302$                  & $ 0.252$                & $ 0.102$                   & $ 0.642$                       \\
            South Africa         & Chlorocebus sabaeus & $ 0.007$              & $ 0.312$                  & $ 0.246$                & $ 0.094$                   & $ 0.638$                       \\
            Saint Kitts          & Chlorocebus sabaeus & $ 0.004$              & $ 0.306$                  & $ 0.250$                & $ 0.106$                   & $ 0.664$                       \\
            Zambia               & Chlorocebus sabaeus & $ 0.006$              & $ 0.308$                  & $ 0.261$                & $ 0.103$                   & $ 0.630$                       \\
            African              & Homo sapiens        & $ 0.003$              & $ 0.256$                  & $ 0.141$                & $ 0.103$                   & $ 1.000$                       \\
            Admixed American     & Homo sapiens        & $ 0.002$              & $ 0.247$                  & $ 0.145$                & $ 0.103$                   & $ 0.945$                       \\
            East Asian           & Homo sapiens        & $ 0.002$              & $ 0.241$                  & $ 0.129$                & $ 0.098$                   & $ 0.918$                       \\
            European             & Homo sapiens        & $ 0.002$              & $ 0.254$                  & $ 0.130$                & $ 0.098$                   & $ 0.919$                       \\
            South Asian          & Homo sapiens        & $ 0.002$              & $ 0.246$                  & $ 0.131$                & $ 0.096$                   & $ 0.912$                       \\
        \end{longtable}
    \end{center}
    \begin{itemize}
        \item Tajima $\theta$ is the synonymous diversity.
        \item $\mathbb{P}_{div}(S > 0)$ is the proportion of substitution in the terminal branch observed with a selection coefficient at the phylogenetic scale ($\Sphy$) larger than 0.
        \item $\mathbb{P}_{obs}[S>0]$ is the proportion of SNPs observed with a predicted selection coefficient at the phylogenetic scale ($\Sphy$) larger than 0.
        \item $\mathbb{P} [ \beta > 0 ]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) larger than 1, estimated for the whole genome.
        \item $\proba [ \polyAdv | \divAdv]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) larger than 1, estimated on the subset of SNPs observed with a predicted selection coefficient at the phylogenetic scale ($\Sphy$) larger than 0.
    \end{itemize}

    \subsection{Correlation with diversity}

    \subsubsection{Proportion of nearly-neutral mutations ($\polyNeutral$)}

    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression/results.watterson.all_P-Seq0.scatter.pdf}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression/results.watterson.neg-weak_P-Seq0.scatter.pdf}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression/results.watterson.pos-weak_P-Seq0.scatter.pdf}
    \end{minipage}

    \begin{itemize}
        \item $\ \proba [ \polyNeutral ]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) between -1 and 1, estimated for the whole genome.
        \item $\ \proba [ \polyNeutral | \divWeakDel]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) between -1 and 1, estimated of SNPs observed with a selection coefficient at the phylogenetic scale ($\Sphy$) between -1 and 0.
        \item $\ \proba [ \polyNeutral | \divWeakAdv ]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) between -1 and 1, estimated of SNPs observed with a selection coefficient at the phylogenetic scale ($\Sphy$) between 0 and 1.
    \end{itemize}

    \subsubsection{Proportion of deleterious mutations ($\polyDel$)}

    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression/results.watterson.all_P-Sinf0.scatter.pdf}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression/results.watterson.neg_P-Sinf0.scatter.pdf}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression/results.watterson.neg-strong_P-Sinf0.scatter.pdf}
    \end{minipage}

    \begin{itemize}
        \item $\ \proba [ \polyDel ]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) lower than -1, estimated for the whole genome.
        \item $\ \proba [ \polyDel | \divDel]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) lower than -1, estimated of SNPs observed with a selection coefficient at the phylogenetic scale ($\Sphy$) between -3 and -3.
        \item $\ \proba [ \polyDel | \divStrongDel ]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) lower than -1, estimated of SNPs observed with a selection coefficient at the phylogenetic scale ($\Sphy$) lower than -3.
    \end{itemize}

    \subsubsection{Proportion of advantageous mutations ($\polyAdv$)}

    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression/results.watterson.all_P-Ssup0.scatter.pdf}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression/results.watterson.pos_P-Ssup0.scatter.pdf}
    \end{minipage}

    \begin{itemize}
        \item $\ \proba [ \polyAdv  ]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) greater than 1, estimated for the whole genome.
        \item $\ \proba [ \polyAdv | \divAdv] $ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) greater than 1, estimated of SNPs observed with a selection coefficient at the phylogenetic scale ($\Sphy$) greater than 1.
    \end{itemize}


    \input{include-MutSel-non-adaptive}


    \section{SIFT scores}
    \input{include-SIFT-non-adaptive}


    \section{Site-specific $\omega$-based codon models.}
    The $61$-by-$61$ codon substitution matrix ($\UniDimArray{q^{(i)}}$) at site $i$ is defined entirely by the mutation matrix ($\UniDimArray{\mu}$), $\omega^{(i)}$ and the genetic code:
    \begin{equation}
        \begin{dcases}
            q_{a \mapsto b}^{(i)} & = 0 \text{ if codons $a$ and $b$ are more than one mutation away,} \\
            q_{a \mapsto b}^{(i)} & = \mu_{a \mapsto b} \text{ if codons $a$ and $b$ are synonymous,} \\
            q_{a \mapsto b}^{(i)} & = \omega^{(i)} \mu_{a \mapsto b} \text{ if codons $a$ and $b$ are non-synonymous}.
        \end{dcases}
    \end{equation}
    By definition of the instantaneous rate matrix, the sum of the entries in each row of the codon substitution rate matrix $\UniDimArray{q}$ is equal to $0$, giving the diagonal entries:
    \begin{equation}
        q_{a \mapsto b}^{(i)}= - \sum\limits_{b \neq a, b = 1}^{61} q_{a \mapsto b}^{(i)}.
    \end{equation}

    In BayesCode (\url{https://github.com/ThibaultLatrille/bayescode}), $\omega$-based site-specific codon models are obtained by running \textit{mutselomega} with the options:
    \begin{scriptsize}
        \begin{verbatim}
        mutselomega --omegashift 0.0 --freeomega --omegancat 30 --flatfitness -a my_alignment.phy -t my_tree.newick -u 2000 my_genename
        \end{verbatim}
    \end{scriptsize}
    The mean value of $\omega$ per site is then obtained by running \textit{readmutselomega} with the options:
    \begin{scriptsize}
        \begin{verbatim}
        readmutselomega --every 1 --until 2000 --burnin 1000 -c 0.025 my_genename
        \end{verbatim}
    \end{scriptsize}

    \input{include-Omega-adaptive}
\end{document}