%TC:ignore
\documentclass{article}
\usepackage{caption}
\usepackage{xcolor, colortbl}
\definecolor{BLUELINK}{HTML}{0645AD}
\definecolor{DARKBLUELINK}{HTML}{0B0080}
\PassOptionsToPackage{hyphens}{url}
\usepackage[colorlinks=false]{hyperref}
% for linking between references, figures, TOC, etc in the pdf document
\hypersetup{colorlinks,
    linkcolor=DARKBLUELINK,
    anchorcolor=DARKBLUELINK,
    citecolor=DARKBLUELINK,
    filecolor=DARKBLUELINK,
    menucolor=DARKBLUELINK,
    urlcolor=BLUELINK
} % Color citation links in purple
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{naturalnames}{hyperref}

\usepackage[margin=50pt]{geometry}
\usepackage{amssymb,amsfonts,amsmath,amsthm,mathtools}
\usepackage{lmodern}
\usepackage{bm,bbold}
\usepackage{verbatim}
\usepackage{float}
\usepackage{listings, enumerate, enumitem}
\usepackage[export]{adjustbox}
\usepackage{tabu}
\usepackage{longtable}
\tabulinesep=0.6mm
\newcommand\cellwidth{\TX@col@width}
\usepackage{hhline}
\setlength{\arrayrulewidth}{1.2pt}
\usepackage{multicol,multirow,array}
\usepackage{etoolbox}
\AtBeginEnvironment{tabu}{\footnotesize}
\usepackage{booktabs}
\usepackage{graphicx}
\pdfinclusioncopyfonts=1

\renewcommand{\thetable}{S\arabic{table}}
\renewcommand{\thefigure}{S\arabic{figure}}

\newcommand{\UniDimArray}[1]{\bm{#1}}
\newcommand{\BiDimArray}[1]{\bm{#1}}
\DeclareMathOperator{\E}{\mathbb{E}}
\DeclareMathOperator{\Var}{\textrm{Var}}
\newcommand{\der}{\textrm{d}}
\newcommand{\e}{\textrm{e}}
\newcommand{\avg}[1]{\left< #1 \right>} % for average
\newcommand{\Ne}{N_{\textrm{e}}}
\newcommand{\proba}{\mathbb{P}}
\newcommand{\pfix}{\proba_{\textrm{fix}}}
\newcommand{\dn}{d_N}
\newcommand{\ds}{d_S}
\newcommand{\dnds}{\dn / \ds}
\newcommand{\Sphy}{S}
\newcommand{\Sphyclass}{\mathcal{C}}
\newcommand{\SphyMean}{\overline{\Sphy}}
\newcommand{\divStrongDel}{\Sphy < -3}
\newcommand{\divDel}{-3 < \Sphy < -1}
\newcommand{\divWeakDel}{-1 < \Sphy < 0}
\newcommand{\divWeakAdv}{0 < \Sphy < 1}
\newcommand{\divAdv}{ \Sphy > 1}
\newcommand{\PdivStrongDel}{\proba \left[ \divStrongDel \right]}
\newcommand{\PdivDel}{\proba \left[ \divDel \right]}
\newcommand{\PdivWeakDel}{\proba \left[ \divWeakDel \right]}
\newcommand{\PdivWeakAdv}{\proba \left[ \divWeakAdv \right]}
\newcommand{\PdivAdv}{\proba \left[ \divAdv \right]}

\newcommand{\Spop}{\beta}
\newcommand{\SpopMean}{\overline{\Spop}}
\newcommand{\polyDel}{\Spop < -1}
\newcommand{\polyNeutral}{-1 < \Spop < 1}
\newcommand{\polyAdv}{ \Spop > 1}
\newcommand{\PpolyDel}{\proba \left[ \polyDel \right]}
\newcommand{\PpolyNeutral}{\proba \left[ \polyNeutral \right]}
\newcommand{\PpolyAdv}{\proba \left[ \polyAdv \right]}

\title{In mammals, 20 to 40\% of advantageous mutations are restoring damaged genomes instead of creating adaptive innovations}

\author{
    \large
    \textbf{T. {Latrille}$^{1}$, J. {Joseph}$^{2}$, N. {Salamin}$^{1}$}\\
    \normalsize $^{1}$Université de Lausanne, Lausanne, Switzerland\\
    \normalsize $^{2}$Université de Lyon, CNRS, LBBE UMR 5558, Villeurbanne, France \\
    \normalsize \texttt{\href{mailto:thibault.latrille@ens-lyon.org}{thibault.latrille@ens-lyon.org}} \\
}

\date{}

\begin{document}
    \maketitle
    \part*{Supplementary materials}
    \tableofcontents
    \clearpage


    \section{Mutation-selection codon models}\label{sec:site-specific-mutation-selection-codon-models}
    In BayesCode (\url{https://github.com/ThibaultLatrille/bayescode}), mutation-selection codon models are obtained by running \textit{mutselomega} for 2000 points of MCMC with the options:
    \begin{scriptsize}
        \begin{verbatim}
        mutselomega ---omegashift 0.0 --ncat 30 -a my_alignment.phy -t my_tree.newick -u 2000 my_genename
        \end{verbatim}
    \end{scriptsize}
    The collection of site-specific fitness profiles ($\UniDimArray{F^{(i)}}, \forall i$) are then obtained by running \textit{readmutselomega}, reading 1000 points of MCMC (first 1000 are considered as burn-in) with the options:
    \begin{scriptsize}
        \begin{verbatim}
        readmutselomega --every 1 --until 2000 --burnin 1000 --ss my_genename
        \end{verbatim}
    \end{scriptsize}
    The gene-specific mutation matrix ($\UniDimArray{\mu}$) is also obtained by running \textit{readmutselomega}, reading 1000 points of MCMC (first 1000 are considered as burn-in) with the options:
    \begin{scriptsize}
        \begin{verbatim}
        readmutselomega --every 1 --until 2000 --burnin 1000 --nuc my_genename
        \end{verbatim}
    \end{scriptsize}

    \subsection{Summary tables}\label{subsec:summary-table-mutsel}
    \begin{center}
        \captionof{table}{~}
        \scriptsize
        \begin{longtable*}{|l|l|r|r|r|r|r|r|}
            \toprule
            Population           & Species             & Tajima $\theta$ & $\mathbb{P} [ \Spop > 1 ]$ & $\mathbb{P}_{mut}[\Sphy > 1]$ & $\mathbb{P} [ \Spop > 1  | \Sphy > 1]$ & $\mathbb{P}_{div}[\Sphy > 1]$ & $\dn(\Sphy > 1) / \ds$ \\
            \midrule
            \endhead
            \midrule
            \multicolumn{8}{r}{{Continued on next page}} \\
            \midrule
            \endfoot

            \bottomrule
            \endlastfoot
            Equus c.             & Equus caballus      & $ 0.002$        & $ 0.042$                   & $ 0.018$                      & $ 0.404$                             & $ 0.115$                      & $ 1.197$           \\
            Iran                 & Bos taurus          & $ 0.005$        & $ 0.036$                   & $ 0.017$                      & $ 0.368$                             & $ 0.125$                      & $ 1.322$           \\
            Uganda               & Bos taurus          & $ 0.006$        & $ 0.035$                   & $ 0.017$                      & $ 0.369$                             & $ 0.126$                      & $ 1.363$           \\
            Australia            & Capra hircus        & $ 0.003$        & $ 0.036$                   & $ 0.017$                      & $ 0.144$                             & $ 0.122$                      & $ 1.035$           \\
            France               & Capra hircus        & $ 0.004$        & $ 0.034$                   & $ 0.017$                      & $ 0.168$                             & $ 0.122$                      & $ 1.043$           \\
            Iran (C. aegagrus)   & Capra hircus        & $ 0.004$        & $ 0.035$                   & $ 0.017$                      & $ 0.093$                             & $ 0.123$                      & $ 1.042$           \\
            Iran                 & Capra hircus        & $ 0.004$        & $ 0.030$                   & $ 0.017$                      & $ 0.140$                             & $ 0.124$                      & $ 1.052$           \\
            Italy                & Capra hircus        & $ 0.004$        & $ 0.032$                   & $ 0.017$                      & $ 0.150$                             & $ 0.122$                      & $ 1.037$           \\
            Morocco              & Capra hircus        & $ 0.004$        & $ 0.029$                   & $ 0.017$                      & $ 0.137$                             & $ 0.124$                      & $ 1.057$           \\
            Iran                 & Ovis aries          & $ 0.008$        & $ 0.022$                   & $ 0.018$                      & $ 0.046$                             & $ 0.104$                      & $ 1.180$           \\
            Iran (O. orientalis) & Ovis aries          & $ 0.009$        & $ 0.025$                   & $ 0.018$                      & $ 0.071$                             & $ 0.103$                      & $ 1.187$           \\
            Iran (O. vignei)     & Ovis aries          & $ 0.008$        & $ 0.028$                   & $ 0.018$                      & $ 0.220$                             & $ 0.104$                      & $ 1.172$           \\
            Various              & Ovis aries          & $ 0.009$        & $ 0.024$                   & $ 0.018$                      & $ 0.108$                             & $ 0.103$                      & $ 1.182$           \\
            Morocco              & Ovis aries          & $ 0.008$        & $ 0.020$                   & $ 0.018$                      & $ 0.070$                             & $ 0.104$                      & $ 1.189$           \\
            Barbados             & Chlorocebus sabaeus & $ 0.004$        & $ 0.040$                   & $ 0.015$                      & $ 0.374$                             & $ 0.123$                      & $ 1.341$           \\
            Central Afr. Rep.    & Chlorocebus sabaeus & $ 0.006$        & $ 0.034$                   & $ 0.015$                      & $ 0.368$                             & $ 0.124$ & $ 1.370$ \\
            Ethiopia             & Chlorocebus sabaeus & $ 0.005$        & $ 0.040$                   & $ 0.015$                      & $ 0.368$                             & $ 0.124$                      & $ 1.361$           \\
            Gambia               & Chlorocebus sabaeus & $ 0.005$        & $ 0.031$                   & $ 0.015$                      & $ 0.377$                             & $ 0.124$                      & $ 1.359$           \\
            Kenya                & Chlorocebus sabaeus & $ 0.006$        & $ 0.035$                   & $ 0.015$                      & $ 0.308$                             & $ 0.125$                      & $ 1.371$           \\
            Nevis                & Chlorocebus sabaeus & $ 0.004$        & $ 0.037$                   & $ 0.015$                      & $ 0.368$                             & $ 0.124$                      & $ 1.349$           \\
            South Africa         & Chlorocebus sabaeus & $ 0.007$        & $ 0.034$                   & $ 0.015$                      & $ 0.368$                             & $ 0.126$                      & $ 1.391$           \\
            Saint Kitts          & Chlorocebus sabaeus & $ 0.004$        & $ 0.039$                   & $ 0.015$                      & $ 0.368$                             & $ 0.125$                      & $ 1.362$           \\
            Zambia               & Chlorocebus sabaeus & $ 0.006$        & $ 0.038$                   & $ 0.015$                      & $ 0.368$                             & $ 0.124$                      & $ 1.362$           \\
            African              & Homo sapiens        & $ 0.003$        & $ 0.038$                   & $ 0.015$                      & $ 0.604$                             & $ 0.104$                      & $ 1.475$           \\
            Admixed American     & Homo sapiens        & $ 0.002$        & $ 0.037$                   & $ 0.015$                      & $ 0.574$                             & $ 0.103$                      & $ 1.466$           \\
            East Asian           & Homo sapiens        & $ 0.002$        & $ 0.036$                   & $ 0.015$                      & $ 0.484$                             & $ 0.104$                      & $ 1.478$           \\
            European             & Homo sapiens        & $ 0.002$        & $ 0.037$                   & $ 0.015$                      & $ 0.517$                             & $ 0.103$                      & $ 1.461$           \\
            South Asian          & Homo sapiens        & $ 0.002$        & $ 0.036$                   & $ 0.015$                      & $ 0.532$                             & $ 0.104$                      & $ 1.473$           \\
        \end{longtable*}
    \end{center}
    \begin{itemize}
        \item Tajima $\theta$ is the synonymous diversity.
        \item $\mathbb{P} [ \Spop > 1 ]$ is the probability of $\Spop > 1$ estimated by polyDFE from all SNPs.
        \item $\mathbb{P}_{mut}[\Sphy > 0]$ is the proportion of mutational opportunities with a predicted selection coefficient larger than 1 ($\Sphy > 1$).
        \item $\mathbb{P} [ \Spop > 1  | \Sphy > 1]$ is the probability of $\Spop > 1$ estimated by polyDFE for SNPs with a predicted selection coefficient larger than 1 ($\Sphy > 1$).
        \item $\mathbb{P}_{div}[\Sphy > 1]$ is the proportion of substitutions in the terminal branch observed with a selection coefficient at the phylogenetic scale larger than 1 ($\Sphy > 1$).
        \item $\dn(\Sphy > 1) / \ds$ is the ratio of non-synonymous over synonymous divergence (also called $\dnds$) estimated for the non-synonymous substitutions in the terminal branch with a selection coefficient larger than 1 ($\Sphy > 1$).
    \end{itemize}

    \newpage
    \begin{center}
        \captionof{table}{~}
        \scriptsize
        \begin{longtable*}{|l|l|r|r|r|r|r|r|r|r|}
            \toprule
            Population           & Species             & Tajima $\theta_{\pi}$ & $\mathbb{P}_{div}[\Sphy > 0]$ & $\dnds $ & $\dn(\Sphy < 0) / \ds$ & R($\dnds $) & $p_+ $ & $p_+(\Sphy < 0)$ & R($p_+ $) \\
            \midrule
            \endhead
            \midrule
            \multicolumn{9}{r}{{Continued on next page}} \\
            \midrule
            \endfoot

            \bottomrule
            \endlastfoot
            Equus c.             & Equus caballus      & $ 0.002$              & $ 0.275$                      & $ 0.148$ & $ 0.113$     & $  23.4$      & $ 0.042$ & $ 0.030$ & $  28.5$ \\
            Iran                 & Bos taurus          & $ 0.005$              & $ 0.274$                      & $ 0.130$ & $ 0.100$     & $  23.3$      & $ 0.036$ & $ 0.025$   & $  30.9$    \\
            Uganda               & Bos taurus          & $ 0.006$              & $ 0.277$                      & $ 0.133$ & $ 0.101$     & $  23.6$      & $ 0.035$ & $ 0.023$   & $  33.7$    \\
            Australia            & Capra hircus        & $ 0.003$              & $ 0.286$                      & $ 0.126$ & $ 0.095$     & $  24.6$      & $ 0.036$ & $ 0.026$ & $  26.8$ \\
            France               & Capra hircus        & $ 0.004$              & $ 0.286$                      & $ 0.127$ & $ 0.096$     & $  24.6$      & $ 0.034$ & $ 0.025$ & $  26.7$ \\
            Iran (C. aegagrus)   & Capra hircus        & $ 0.004$              & $ 0.285$                      & $ 0.126$ & $ 0.095$     & $  24.5$      & $ 0.035$ & $ 0.026$ & $  25.0$ \\
            Iran                 & Capra hircus        & $ 0.004$              & $ 0.288$                      & $ 0.126$ & $ 0.095$     & $  24.8$      & $ 0.030$ & $ 0.020$   & $  31.8$    \\
            Italy                & Capra hircus        & $ 0.004$              & $ 0.287$                      & $ 0.127$ & $ 0.096$     & $  24.6$      & $ 0.032$ & $ 0.022$   & $  30.6$    \\
            Morocco              & Capra hircus        & $ 0.004$              & $ 0.288$                      & $ 0.126$ & $ 0.095$     & $  24.7$      & $ 0.029$ & $ 0.020$ & $  31.5$ \\
            Iran                 & Ovis aries          & $ 0.008$              & $ 0.243$                      & $ 0.149$ & $ 0.120$     & $  20.0$      & $ 0.022$ & $ 0.014$   & $  36.5$    \\
            Iran (O. orientalis) & Ovis aries          & $ 0.009$              & $ 0.241$                      & $ 0.151$ & $ 0.121$     & $  19.8$      & $ 0.025$ & $ 0.016$ & $  33.4$ \\
            Iran (O. vignei)     & Ovis aries          & $ 0.008$              & $ 0.243$                      & $ 0.149$ & $ 0.119$     & $  20.0$      & $ 0.028$ & $ 0.019$ & $  30.5$ \\
            Various              & Ovis aries          & $ 0.009$              & $ 0.240$                      & $ 0.151$ & $ 0.121$     & $  19.7$      & $ 0.024$ & $ 0.015$   & $  36.7$    \\
            Morocco              & Ovis aries          & $ 0.008$              & $ 0.241$                      & $ 0.151$ & $ 0.121$     & $  19.7$      & $ 0.020$ & $ 0.013$   & $  38.0$    \\
            Barbados             & Chlorocebus sabaeus & $ 0.004$              & $ 0.289$                      & $ 0.141$ & $ 0.106$     & $  25.1$      & $ 0.040$ & $ 0.030$ & $  25.3$ \\
            Central Afr. Rep.    & Chlorocebus sabaeus & $ 0.006$              & $ 0.287$                      & $ 0.141$ & $ 0.106$     & $  24.9$ & $ 0.034$ & $ 0.026$ & $  24.7$ \\
            Ethiopia             & Chlorocebus sabaeus & $ 0.005$              & $ 0.290$                      & $ 0.141$ & $ 0.105$     & $  25.2$      & $ 0.040$ & $ 0.030$ & $  25.0$ \\
            Gambia               & Chlorocebus sabaeus & $ 0.005$              & $ 0.287$                      & $ 0.141$ & $ 0.106$     & $  25.0$      & $ 0.031$ & $ 0.021$ & $  31.8$ \\
            Kenya                & Chlorocebus sabaeus & $ 0.006$              & $ 0.290$                      & $ 0.142$ & $ 0.106$     & $  25.2$      & $ 0.035$ & $ 0.026$ & $  24.9$ \\
            Nevis                & Chlorocebus sabaeus & $ 0.004$              & $ 0.289$                      & $ 0.141$ & $ 0.106$     & $  25.1$      & $ 0.037$ & $ 0.028$ & $  23.6$ \\
            South Africa         & Chlorocebus sabaeus & $ 0.007$              & $ 0.291$                      & $ 0.141$ & $ 0.105$     & $  25.4$      & $ 0.035$ & $ 0.025$ & $  27.7$ \\
            Saint Kitts          & Chlorocebus sabaeus & $ 0.004$              & $ 0.290$                      & $ 0.141$ & $ 0.105$     & $  25.3$      & $ 0.039$ & $ 0.029$ & $  26.1$ \\
            Zambia               & Chlorocebus sabaeus & $ 0.006$              & $ 0.288$                      & $ 0.141$ & $ 0.106$     & $  25.0$      & $ 0.038$ & $ 0.029$ & $  25.3$ \\
            African              & Homo sapiens        & $ 0.003$              & $ 0.243$                      & $ 0.195$ & $ 0.156$     & $  20.2$      & $ 0.038$ & $ 0.026$ & $  32.4$ \\
            Admixed American     & Homo sapiens        & $ 0.002$              & $ 0.242$                      & $ 0.195$ & $ 0.156$     & $  20.2$      & $ 0.038$ & $ 0.026$ & $  31.3$ \\
            East Asian           & Homo sapiens        & $ 0.002$              & $ 0.244$                      & $ 0.195$ & $ 0.155$     & $  20.4$      & $ 0.036$ & $ 0.025$ & $  30.2$ \\
            European             & Homo sapiens        & $ 0.002$              & $ 0.242$                      & $ 0.195$ & $ 0.156$     & $  20.1$      & $ 0.036$ & $ 0.025$ & $  30.9$ \\
            South Asian          & Homo sapiens        & $ 0.002$              & $ 0.244$                      & $ 0.195$ & $ 0.156$     & $  20.3$      & $ 0.035$ & $ 0.024$ & $  31.7$ \\
        \end{longtable*}
    \end{center}
    \begin{itemize}
        \item Tajima $\theta$ is the synonymous diversity.
        \item $\mathbb{P}_{div}[\Sphy > 0]$ is the proportion of substitutions in the terminal branch observed with a selection coefficient at the phylogenetic scale larger than 0 ($\Sphy > 0$).
        \item $\dnds$ is the ratio of non-synonymous over synonymous divergence estimated for all the non-synonymous substitutions in the terminal branch.
        \item $\dn(\Sphy < 0) / \ds$ is the ratio of non-synonymous over synonymous divergence estimated for the non-synonymous substitutions in the terminal branch with a selection coefficient lower than 0 ($\Sphy < 0$).
        This is the estimated divergence when we removed the restoring substitutions.
        \item $R(\dnds)=\dfrac{\dnds - \dn(\Sphy < 0) / \ds}{\dnds}$ is the fraction of the divergence ($\dnds$) that is over-estimated, $\dnds$ is compared to the estimated divergence when we removed the restoring substitutions $\dn(\Sphy < 0) / \ds$.
        \item $p_+=\PpolyAdv$ is the probability for a new mutation to be advantageous, estimated (with polyDFE) using all available polymorphism.
        \item $p_+(\Sphy < 0)$ is the probability for a new mutation to be advantageous, estimated (with polyDFE) using polymorphism with a selection coefficient lower than 0 ($\Sphy < 0$).
        This is the estimated probability for a new mutation to be advantageous when we removed the restoring polymorphism from the dataset.
        \item $R(p_+)=\dfrac{p_+ - p_+(\Sphy < 0)}{p_+}$ is the fraction of the probability for a new mutation to be advantageous ($p_+$) that is over-estimated, $p_+$ is compared to its estimation when we had removed all the restoring mutations $p_+(\Sphy < 0)$.
    \end{itemize}

    We estimated that between 20 and 25\% of $\dnds$ is over-estimated ($R(\dnds)$), corresponding to maintenance positive selection exerted on substitutions.
    Moreover, we estimated that between 24 and 38\% of $p_+$ is over-estimated ($R(p_+)$), corresponding to maintenance positive selection exerted on segregating mutations.
    Altogether, in a nearly neutral framework, genomes could be viewed as the arena of a tug-of-war between a large volume of deleterious mutations that eventually reach fixation due to drift, and scarce advantageous mutations that are restoring a more optimal ancestral state.

    \newpage

    \subsection{Correlation with diversity}

    \subsubsection{Proportion of nearly-neutral mutations ($\polyNeutral$)}\label{subsec:proportion-nearly-neutral-mutations}
    \captionof{figure}{~}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression-MutSel/results.watterson.all_P-Seq0.scatter.pdf}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression-MutSel/results.watterson.neg-weak_P-Seq0.scatter.pdf}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression-MutSel/results.watterson.pos-weak_P-Seq0.scatter.pdf}
    \end{minipage}
    \begin{itemize}
        \item $\ \proba [ \polyNeutral ]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) between -1 and 1, estimated for the whole genome.
        \item $\ \proba [ \polyNeutral | \divWeakDel]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) between -1 and 1, estimated of SNPs observed with a selection coefficient at the phylogenetic scale ($\Sphy$) between -1 and 0.
        \item $\ \proba [ \polyNeutral | \divWeakAdv ]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) between -1 and 1, estimated of SNPs observed with a selection coefficient at the phylogenetic scale ($\Sphy$) between 0 and 1.
    \end{itemize}

    \subsubsection{Proportion of deleterious mutations ($\polyDel$)}\label{subsec:proportion-deleterious-mutations}
    \captionof{figure}{~}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression-MutSel/results.watterson.all_P-Sinf0.scatter.pdf}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression-MutSel/results.watterson.neg_P-Sinf0.scatter.pdf}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression-MutSel/results.watterson.neg-strong_P-Sinf0.scatter.pdf}
    \end{minipage}
    \begin{itemize}
        \item $\ \proba [ \polyDel ]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) lower than -1, estimated for the whole genome.
        \item $\ \proba [ \polyDel | \divDel]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) lower than -1, estimated of SNPs observed with a selection coefficient at the phylogenetic scale ($\Sphy$) between -3 and -3.
        \item $\ \proba [ \polyDel | \divStrongDel ]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) lower than -1, estimated of SNPs observed with a selection coefficient at the phylogenetic scale ($\Sphy$) lower than -3.
    \end{itemize}

    \subsubsection{Proportion of advantageous mutations ($\polyAdv$)}\label{subsec:proportion-advantageous-mutations}
    \captionof{figure}{~}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression-MutSel/results.watterson.all_P-Ssup0.scatter.pdf}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{../experiments/5bins-polyDFE-mC/regression-MutSel/results.watterson.pos_P-Ssup0.scatter.pdf}
    \end{minipage}
    \begin{itemize}
        \item $\ p_+ = \proba [ \polyAdv  ]$ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) greater than 1, estimated for the whole genome.
        \item $\ p_+( \divAdv) = \proba [ \polyAdv | \divAdv] $ is the proportion of DFE with a selection coefficient at the population-genetic scale ($\Spop$) greater than 1, estimated of SNPs observed with a selection coefficient at the phylogenetic scale ($\Sphy$) greater than 1.
    \end{itemize}

    \subsection{Selection efficacy correlation with diversity}\label{subsec:selection-efficacy-correlation-with-diversity}
    The mean of the DFE $\SpopMean$ estimated by polyDFE is given by:
    \begin{align*}
        \SpopMean & = \int_{-\infty}^{+\infty} \Spop \phi \left( \Spop; \Spop_d , b, p_b, \Spop_b \right) \der \Spop \\
        & =  p_b \Spop_b + \left( 1 - p_b \right) \Spop_d
    \end{align*}

    To look further into the relation between diversity and selection efficacy, we studied the estimation of the mean $\SpopMean$ as a function of the mean $\SphyMean$ for a subset of SNPs, for which $\SphyMean$ is close to $0$.
    Indeed, polyDFE is a nearly-neutral model and is not able to reliably compute selection coefficients for only strongly deleterious or only strongly advantageous SNPs.
    Sliding windows of $5,000$ SNPs are taken with a mean $\SphyMean$ between $-0.5$ and $0.5$.
    $\SpopMean$ is estimated for each group, with some groups that might have part of same SNPs that overlap.
    The relationship between $\SpopMean$ and $\SphyMean$ can be described by a second-order polynomial, and the slope of this polynomial at $\SphyMean=0$ gives the ratio $\SpopMean/\SphyMean$ for each population.
    \captionof{figure}{~}
    \begin{center}
        \includegraphics[width=\linewidth, page=1]{artworks/figure.slope-beta-S}
    \end{center}

    We observe that $\SpopMean/\SphyMean$, representing the efficacy of selection at the short versus long-term is positively correlated with synonymous diversity, a proxy of effective population size.
    Altogether, populations with a large effective population size are under more efficient selection, confirming the prediction of the nearly-neutral theory of evolution that mutations are less likely to be considered neutral in populations with high diversity (see~\ref{subsec:proportion-nearly-neutral-mutations}, left scatter-plot), especially for mutations predicted already to be nearly-neutral at the phylogenetic scale (see~\ref{subsec:proportion-nearly-neutral-mutations}, middle and right scatter-plot).
    As a consequence of an increased selection efficiency, mutations in populations with high diversity are mechanistically more likely to be deleterious (see~\ref{subsec:proportion-deleterious-mutations}, left scatter-plot), especially for mutations predicted already to be deleterious at the phylogenetic scale (see~\ref{subsec:proportion-deleterious-mutations}, middle and right scatter-plot).
    % A decrease in the proportion of neutral mutation did not translate into a mechanical increase in the proportion of advantageous mutations.
    % Instead, populations with higher diversity have a lower proportion of advantageous mutations, this effect is also more pronounced for mutations predicted to be advantageous at the phylogenetic scale.
    % This discrepancy can be explained by the fact that populations with larger diversity have already depleted the putatively advantageous mutations, and that only deleterious are remaining possible.
    % In other words, being on the top of a fixed fitness landscape leaves you with only paths going downhill.
    % Altogether, populations with increased diversity are showing signature of increased selection efficiency and are also less likely to generate positively selected mutations.

    \newpage

    \subsection{Proportion of selected mutations}

    For each population, the proportion of advantageous ($p_+=\PpolyAdv$), nearly-neutral ($\PpolyNeutral$) and of deleterious ($\PpolyDel$) mutations estimated at the population-genetic scale is shown.
    Estimation is performed with SNPs with different class of selection at the phylogenetic scale: $\Sphyclass \in \{ \divStrongDel, \divDel,  \divWeakDel,  \divWeakAdv, \divAdv \}$
    Population are sorted for increased synonymous diversity.

    \begin{center}
        \captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.MutSel.all.stacked.pdf}
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}
        \captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.MutSel.neg-strong.stacked.pdf} \\
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}\captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.MutSel.neg.stacked.pdf} \\
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}\captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.MutSel.neg-weak.stacked.pdf} \\
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}\captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.MutSel.pos-weak.stacked.pdf} \\
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}\captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.MutSel.pos.stacked.pdf}
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}
    \end{center}
    We confirmed that higher synonymous diversity is typically accompanied by a smaller proportion of neutral mutations.
    This result is specially visible when we restricted the analysis to class of mutations that are supposedly nearly-neutral at the phylogenetic scale ($\divWeakDel$  and $\divWeakAdv$).
    This result suggests that populations with higher diversity (e.g.~\textit{Bos} or \textit{Ovis}) are more likely to discriminate whether mutations are advantageous or deleterious.
    Alternatively stated, mutations in populations with low diversity (e.g.~\textit{Homo}) are effectively nearly-neutral and behave as would a neutral mutation.
    This result is qualitatively in accordance with the nearly-neutral theory of evolution which argues that mutations are less efficiently selected for in small populations.

    \section{Mutation-selection models: histograms and SFS for each population}\label{sec:mutation-selection-models:-histograms-and-sfs-for-each-population}
    For each population, the same figure is shown.
    \begin{itemize}
        \item Top-left panel: Histogram of predicted selection coefficients ($\Sphy$) for all possible mutations away from the ancestral genome, giving the number of sites ($L$, in y-axis) across the genome with a given $\Sphy$.
        Mutations are divided into 5 classes: severely deleterious (blue), deleterious (green), weakly deleterious (light green), weakly advantageous (yellow) and advantageous (red).
        \item Top-right panel: Histogram of predicted selection coefficients ($\Sphy$) for all observed mutations in a sample of 8 individuals (out of 512 in the original dataset) of African descent.
        If they are less mutations observed than expected, this class is thus undergoing purifying selection.
        \item Bottom-left panel: The site-frequency spectrum (SFS) represents the proportion of mutations (y-axis) with a given number of derived alleles in the population (x-axis).
        SFS are drawn for a random sample of 16 alleles (mean in solid line and standard deviation in filled color) for each class of selection coefficient and for synonymous mutations which are supposedly neutral (black).
        At high frequencies, supposedly severally deleterious mutations are underrepresented.
        \item Bottom-right panel:For each class of selection (and for the set of all non-synonymous mutations), information from the SFS and the expected total mutation rate are combined at the population-genetic scale to estimate the proportion of advantageous mutations $\PpolyAdv$, of nearly-neutral mutations $\PpolyNeutral$ and of deleterious mutations $\PpolyDel$.
    \end{itemize}

    \input{include-MutSel-non-adaptive}

    \newpage
    \section{SIFT scores}\label{sec:sift-scores}

    \subsection{Proportion of selected mutations}\label{subsec:proportion-of-selected-mutations}
    For each population, the proportion of advantageous ($p_+=\PpolyAdv$), nearly-neutral ($\PpolyNeutral$) and of deleterious ($\PpolyDel$) mutations estimated at the population-genetic scale is shown.
    Estimation is performed with SNPs with different class of SIFT scores at the phylogenetic scale.
    Population are sorted for increased synonymous diversity.

    \begin{center}
        \captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.SIFT.all.stacked.pdf}
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}
        \captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.SIFT.neg-strong.stacked.pdf} \\
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}\captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.SIFT.neg.stacked.pdf} \\
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}\captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.SIFT.neg-weak.stacked.pdf} \\
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}\captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.SIFT.pos-weak.stacked.pdf} \\
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}\captionof{figure}{~}
        \begin{minipage}{0.9\linewidth}
            \includegraphics[width=0.95\linewidth, page=1]{../experiments/5bins-polyDFE-mC/results/Theta.SIFT.pos.stacked.pdf}
        \end{minipage}
        \begin{minipage}{0.09\linewidth}
            \includegraphics[width=\linewidth, page=1]{artworks/legend.polycat}
        \end{minipage}
    \end{center}
    Sites with a higher SIFT score at the phylogenetic scale are more positively selected in the different populations.

    \subsection{Summary table}\label{subsec:summary-table-sift-scores}
    \begin{center}
        \captionof{table}{~}
        \footnotesize
        \begin{longtable}{|l|l|r|r|r|r|}
            \toprule
            Population           & Species             & Tajima $\theta_{\pi}$ & $p_+$    & $p_+ (\textrm{SIFT} < 0.8)$ & R($p_+$) \\
            \midrule
            \endhead
            \midrule
            \multicolumn{6}{r}{{Continued on next page}} \\
            \midrule
            \endfoot

            \bottomrule
            \endlastfoot
            Equus c.             & Equus caballus      & $ 0.002$              & $ 0.042$ & $ 0.037$                    & $  12.2$   \\
            Iran                 & Bos taurus          & $ 0.005$              & $ 0.036$ & $ 0.031$                    & $  14.7$   \\
            Uganda               & Bos taurus          & $ 0.006$              & $ 0.035$ & $ 0.029$                    & $  16.2$   \\
            Australia            & Capra hircus        & $ 0.003$              & $ 0.036$ & $ 0.032$                    & $ 9.214$ \\
            France               & Capra hircus        & $ 0.004$              & $ 0.034$ & $ 0.034$                    & $-1.445$ \\
            Iran (C. aegagrus)   & Capra hircus        & $ 0.004$              & $ 0.035$ & $ 0.032$                    & $ 9.881$ \\
            Iran                 & Capra hircus        & $ 0.004$              & $ 0.030$ & $ 0.026$                    & $  12.3$   \\
            Italy                & Capra hircus        & $ 0.004$              & $ 0.032$ & $ 0.029$                    & $ 8.754$ \\
            Morocco              & Capra hircus        & $ 0.004$              & $ 0.029$ & $ 0.025$                    & $  12.7$   \\
            Iran                 & Ovis aries          & $ 0.008$              & $ 0.022$ & $ 0.019$                    & $  12.9$   \\
            Iran (O. orientalis) & Ovis aries          & $ 0.009$              & $ 0.025$ & $ 0.021$                    & $  13.6$   \\
            Iran (O. vignei)     & Ovis aries          & $ 0.008$              & $ 0.028$ & $ 0.024$                    & $  14.2$   \\
            Various              & Ovis aries          & $ 0.009$              & $ 0.024$ & $ 0.019$                    & $  19.0$   \\
            Morocco              & Ovis aries          & $ 0.008$              & $ 0.020$ & $ 0.017$                    & $  14.4$   \\
            Barbados             & Chlorocebus sabaeus & $ 0.004$              & $ 0.040$ & $ 0.037$                    & $ 7.660$ \\
            Central Afr. Rep.    & Chlorocebus sabaeus & $ 0.006$              & $ 0.034$ & $ 0.032$                    & $ 7.610$ \\
            Ethiopia             & Chlorocebus sabaeus & $ 0.005$              & $ 0.040$ & $ 0.037$                    & $ 7.464$ \\
            Gambia               & Chlorocebus sabaeus & $ 0.005$              & $ 0.031$ & $ 0.028$                    & $ 9.891$ \\
            Kenya                & Chlorocebus sabaeus & $ 0.006$              & $ 0.035$ & $ 0.033$                    & $ 6.180$ \\
            Nevis                & Chlorocebus sabaeus & $ 0.004$              & $ 0.037$ & $ 0.035$                    & $ 6.532$ \\
            South Africa         & Chlorocebus sabaeus & $ 0.007$              & $ 0.034$ & $ 0.031$                    & $ 8.924$ \\
            Saint Kitts          & Chlorocebus sabaeus & $ 0.004$              & $ 0.039$ & $ 0.036$                    & $ 7.642$ \\
            Zambia               & Chlorocebus sabaeus & $ 0.006$              & $ 0.039$ & $ 0.035$                    & $ 8.197$ \\
            African              & Homo sapiens        & $ 0.003$              & $ 0.038$ & $ 0.032$                    & $  17.1$   \\
            Admixed American     & Homo sapiens        & $ 0.002$              & $ 0.039$ & $ 0.033$                    & $  14.9$   \\
            East Asian           & Homo sapiens        & $ 0.002$              & $ 0.036$ & $ 0.033$                    & $ 9.250$ \\
            European             & Homo sapiens        & $ 0.002$              & $ 0.036$ & $ 0.033$                    & $ 8.704$ \\
            South Asian          & Homo sapiens        & $ 0.002$              & $ 0.036$ & $ 0.032$                    & $ 9.853$ \\
        \end{longtable}

    \end{center}
    \begin{itemize}
        \item Tajima $\theta$ is the synonymous diversity.
        \item $p_+=\PpolyAdv$ is the probability for a new mutation to be advantageous, estimated (with polyDFE) using all available polymorphism.
        \item $p_+ (\textrm{SIFT} < 0.8)$ is the probability for a new mutation to be advantageous, estimated (with polyDFE) using polymorphism with a SIFT lower than 0.8 ($SIFT < 0.8$).
        This is the estimated probability for a new mutation to be advantageous when we removed the restoring polymorphism from the dataset.
        \item $R(p_+)=\dfrac{p_+ - p_+ (\textrm{SIFT} < 0.8)}{p_+}$ is the fraction of the probability for a new mutation to be advantageous ($p_+$) that is over-estimated, $p_+$ is compared to its estimation when we had removed all the restoring mutations $p_+ (\textrm{SIFT} < 0.8)$.
    \end{itemize}

    The previous section showed that mutations with a higher SIFT score at the phylogenetic scale are more positively selected in the different populations.
    However, whenever we removed the category of SNPs with the highest SIFT score ($\textrm{SIFT} \geq 0.8$), we expected the estimation of $p_+ (\textrm{SIFT} < 0.8)$ to be lower than $p_+$ (estimated on the all dataset), as we have shown that $p_+ (\Sphy < 0) < p_+$ with mutation-selection codon models.
    Thus SIFT scores are not providing reliable information on the current selective strength exerted on mutations, at least for high SIFT scores.
    Moreover, there is no defined threshold (here we arbitrarily used 0.8) to cut-off between advantageous and deleterious changes.

    \newpage
    \section{SIFT scores: histograms and SFS for each population}\label{sec:sift-scores:-histograms-and-sfs-for-each-population}
    For each population, the same figure is shown.
    \begin{itemize}
        \item Top-left panel: Histogram of SIFT score for all possible mutations away from the ancestral genome, giving the number of sites ($L$, in y-axis) across the genome with a given SIFT score.
        Mutations are divided into 5 classes: severely deleterious (blue), deleterious (green), weakly deleterious (light green), weakly advantageous (yellow) and advantageous (red).
        \item Top-right panel: Histogram of SIFT score for all observed mutations in a sample of 8 individuals (out of 512 in the original dataset) of African descent.
        If they are less mutations observed than expected, this class is thus undergoing purifying selection.
        \item Bottom-left panel: The site-frequency spectrum (SFS) represents the proportion of mutations (y-axis) with a given number of derived alleles in the population (x-axis).
        SFS are drawn for a random sample of 16 alleles (mean in solid line and standard deviation in filled color) for each class of selection coefficient and for synonymous mutations which are supposedly neutral (black).
        At high frequencies, supposedly severally deleterious mutations are underrepresented.
        \item Bottom-right panel:For each class of selection (and for the set of all non-synonymous mutations), information from the SFS and the expected total mutation rate are combined at the population-genetic scale to estimate the proportion of advantageous mutations $\PpolyAdv$, of nearly-neutral mutations $\PpolyNeutral$ and of deleterious mutations $\PpolyDel$.
    \end{itemize}
    \input{include-SIFT-non-adaptive}

\end{document}